// react-dropdown-select/dist/react-dropdown-select.es.js
var Yt = Object.defineProperty;
var Bt = (e, r, t) => r in e ? Yt(e, r, { enumerable: !0, configurable: !0, writable: !0, value: t }) : e[r] = t;
var F = (e, r, t) => Bt(e, typeof r != "symbol" ? r + "" : r, t);
import * as ie from "react";
import Ee, { forwardRef as zt, useContext as Wt, Component as wr } from "react";
import Ut from "react-dom";
function Ht(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var hr = { exports: {} }, Pe = {};
/** @license React v16.14.0
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ir;
function Kt() {
  if (Ir) return Pe;
  Ir = 1;
  var e = Ee, r = 60103;
  if (Pe.Fragment = 60107, typeof Symbol == "function" && Symbol.for) {
    var t = Symbol.for;
    r = t("react.element"), Pe.Fragment = t("react.fragment");
  }
  var n = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, i = Object.prototype.hasOwnProperty, o = { key: !0, ref: !0, __self: !0, __source: !0 };
  function s(c, d, y) {
    var u, v = {}, E = null, _ = null;
    y !== void 0 && (E = "" + y), d.key !== void 0 && (E = "" + d.key), d.ref !== void 0 && (_ = d.ref);
    for (u in d) i.call(d, u) && !o.hasOwnProperty(u) && (v[u] = d[u]);
    if (c && c.defaultProps) for (u in d = c.defaultProps, d) v[u] === void 0 && (v[u] = d[u]);
    return { $$typeof: r, type: c, key: E, ref: _, props: v, _owner: n.current };
  }
  return Pe.jsx = s, Pe.jsxs = s, Pe;
}
var ar = {};
/** @license React v16.14.0
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var jr;
function Gt() {
  return jr || (jr = 1, function(e) {
    process.env.NODE_ENV !== "production" && function() {
      var r = Ee, t = 60103, n = 60106;
      e.Fragment = 60107;
      var i = 60108, o = 60114, s = 60109, c = 60110, d = 60112, y = 60113, u = 60120, v = 60115, E = 60116, _ = 60121, S = 60122, m = 60117, x = 60129, T = 60131;
      if (typeof Symbol == "function" && Symbol.for) {
        var R = Symbol.for;
        t = R("react.element"), n = R("react.portal"), e.Fragment = R("react.fragment"), i = R("react.strict_mode"), o = R("react.profiler"), s = R("react.provider"), c = R("react.context"), d = R("react.forward_ref"), y = R("react.suspense"), u = R("react.suspense_list"), v = R("react.memo"), E = R("react.lazy"), _ = R("react.block"), S = R("react.server.block"), m = R("react.fundamental"), R("react.scope"), R("react.opaque.id"), x = R("react.debug_trace_mode"), R("react.offscreen"), T = R("react.legacy_hidden");
      }
      var N = typeof Symbol == "function" && Symbol.iterator, W = "@@iterator";
      function p(a) {
        if (a === null || typeof a != "object")
          return null;
        var h = N && a[N] || a[W];
        return typeof h == "function" ? h : null;
      }
      var K = r.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
      function O(a) {
        {
          for (var h = arguments.length, b = new Array(h > 1 ? h - 1 : 0), D = 1; D < h; D++)
            b[D - 1] = arguments[D];
          Oe("error", a, b);
        }
      }
      function Oe(a, h, b) {
        {
          var D = K.ReactDebugCurrentFrame, X = "";
          if (de) {
            var J = Q(de.type), H = de._owner;
            X += Ie(J, de._source, H && Q(H.type));
          }
          X += D.getStackAddendum(), X !== "" && (h += "%s", b = b.concat([X]));
          var I = b.map(function(oe) {
            return "" + oe;
          });
          I.unshift("Warning: " + h), Function.prototype.apply.call(console[a], console, I);
        }
      }
      var he = !1;
      function ve(a) {
        return !!(typeof a == "string" || typeof a == "function" || a === e.Fragment || a === o || a === x || a === i || a === y || a === u || a === T || he || typeof a == "object" && a !== null && (a.$$typeof === E || a.$$typeof === v || a.$$typeof === s || a.$$typeof === c || a.$$typeof === d || a.$$typeof === m || a.$$typeof === _ || a[0] === S));
      }
      var _e = /^(.*)[\\\/]/;
      function Ie(a, h, b) {
        var D = "";
        if (h) {
          var X = h.fileName, J = X.replace(_e, "");
          if (/^index\./.test(J)) {
            var H = X.match(_e);
            if (H) {
              var I = H[1];
              if (I) {
                var oe = I.replace(_e, "");
                J = oe + "/" + J;
              }
            }
          }
          D = " (at " + J + ":" + h.lineNumber + ")";
        } else b && (D = " (created by " + b + ")");
        return `
    in ` + (a || "Unknown") + D;
      }
      var xe = 1;
      function je(a) {
        return a._status === xe ? a._result : null;
      }
      function le(a, h, b) {
        var D = h.displayName || h.name || "";
        return a.displayName || (D !== "" ? b + "(" + D + ")" : b);
      }
      function Q(a) {
        if (a == null)
          return null;
        if (typeof a.tag == "number" && O("Received an unexpected object in getComponentName(). This is likely a bug in React. Please file an issue."), typeof a == "function")
          return a.displayName || a.name || null;
        if (typeof a == "string")
          return a;
        switch (a) {
          case e.Fragment:
            return "Fragment";
          case n:
            return "Portal";
          case o:
            return "Profiler";
          case i:
            return "StrictMode";
          case y:
            return "Suspense";
          case u:
            return "SuspenseList";
        }
        if (typeof a == "object")
          switch (a.$$typeof) {
            case c:
              return "Context.Consumer";
            case s:
              return "Context.Provider";
            case d:
              return le(a, a.render, "ForwardRef");
            case v:
              return Q(a.type);
            case _:
              return Q(a.render);
            case E: {
              var h = a, b = je(h);
              if (b)
                return Q(b);
              break;
            }
          }
        return null;
      }
      var Ce = {};
      K.ReactDebugCurrentFrame;
      var de = null;
      function f(a) {
        de = a;
      }
      function w(a, h, b, D, X) {
        {
          var J = Function.call.bind(Object.prototype.hasOwnProperty);
          for (var H in a)
            if (J(a, H)) {
              var I = void 0;
              try {
                if (typeof a[H] != "function") {
                  var oe = Error((D || "React class") + ": " + b + " type `" + H + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof a[H] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                  throw oe.name = "Invariant Violation", oe;
                }
                I = a[H](h, H, D, b, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
              } catch (me) {
                I = me;
              }
              I && !(I instanceof Error) && (f(X), O("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", D || "React class", b, H, typeof I), f(null)), I instanceof Error && !(I.message in Ce) && (Ce[I.message] = !0, f(X), O("Failed %s type: %s", b, I.message), f(null));
            }
        }
      }
      var P = K.ReactCurrentOwner, A = Object.prototype.hasOwnProperty, k = {
        key: !0,
        ref: !0,
        __self: !0,
        __source: !0
      }, M, $, C;
      C = {};
      function L(a) {
        if (A.call(a, "ref")) {
          var h = Object.getOwnPropertyDescriptor(a, "ref").get;
          if (h && h.isReactWarning)
            return !1;
        }
        return a.ref !== void 0;
      }
      function Y(a) {
        if (A.call(a, "key")) {
          var h = Object.getOwnPropertyDescriptor(a, "key").get;
          if (h && h.isReactWarning)
            return !1;
        }
        return a.key !== void 0;
      }
      function U(a, h) {
        if (typeof a.ref == "string" && P.current && h && P.current.stateNode !== h) {
          var b = Q(P.current.type);
          C[b] || (O('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', Q(P.current.type), a.ref), C[b] = !0);
        }
      }
      function ue(a, h) {
        {
          var b = function() {
            M || (M = !0, O("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", h));
          };
          b.isReactWarning = !0, Object.defineProperty(a, "key", {
            get: b,
            configurable: !0
          });
        }
      }
      function se(a, h) {
        {
          var b = function() {
            $ || ($ = !0, O("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", h));
          };
          b.isReactWarning = !0, Object.defineProperty(a, "ref", {
            get: b,
            configurable: !0
          });
        }
      }
      var g = function(a, h, b, D, X, J, H) {
        var I = {
          // This tag allows us to uniquely identify this as a React Element
          $$typeof: t,
          // Built-in properties that belong on the element
          type: a,
          key: h,
          ref: b,
          props: H,
          // Record the component responsible for creating this element.
          _owner: J
        };
        return I._store = {}, Object.defineProperty(I._store, "validated", {
          configurable: !1,
          enumerable: !1,
          writable: !0,
          value: !1
        }), Object.defineProperty(I, "_self", {
          configurable: !1,
          enumerable: !1,
          writable: !1,
          value: D
        }), Object.defineProperty(I, "_source", {
          configurable: !1,
          enumerable: !1,
          writable: !1,
          value: X
        }), Object.freeze && (Object.freeze(I.props), Object.freeze(I)), I;
      };
      function Le(a, h, b, D, X) {
        {
          var J, H = {}, I = null, oe = null;
          b !== void 0 && (I = "" + b), Y(h) && (I = "" + h.key), L(h) && (oe = h.ref, U(h, X));
          for (J in h)
            A.call(h, J) && !k.hasOwnProperty(J) && (H[J] = h[J]);
          if (a && a.defaultProps) {
            var me = a.defaultProps;
            for (J in me)
              H[J] === void 0 && (H[J] = me[J]);
          }
          if (I || oe) {
            var Re = typeof a == "function" ? a.displayName || a.name || "Unknown" : a;
            I && ue(H, Re), oe && se(H, Re);
          }
          return g(a, I, oe, X, D, P.current, H);
        }
      }
      var Se = K.ReactCurrentOwner;
      K.ReactDebugCurrentFrame;
      function be(a) {
        de = a;
      }
      var nr;
      nr = !1;
      function or(a) {
        return typeof a == "object" && a !== null && a.$$typeof === t;
      }
      function Ar() {
        {
          if (Se.current) {
            var a = Q(Se.current.type);
            if (a)
              return `

Check the render method of \`` + a + "`.";
          }
          return "";
        }
      }
      function Dt(a) {
        return "";
      }
      var kr = {};
      function It(a) {
        {
          var h = Ar();
          if (!h) {
            var b = typeof a == "string" ? a : a.displayName || a.name;
            b && (h = `

Check the top-level render call using <` + b + ">.");
          }
          return h;
        }
      }
      function $r(a, h) {
        {
          if (!a._store || a._store.validated || a.key != null)
            return;
          a._store.validated = !0;
          var b = It(h);
          if (kr[b])
            return;
          kr[b] = !0;
          var D = "";
          a && a._owner && a._owner !== Se.current && (D = " It was passed a child from " + Q(a._owner.type) + "."), be(a), O('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', b, D), be(null);
        }
      }
      function Nr(a, h) {
        {
          if (typeof a != "object")
            return;
          if (Array.isArray(a))
            for (var b = 0; b < a.length; b++) {
              var D = a[b];
              or(D) && $r(D, h);
            }
          else if (or(a))
            a._store && (a._store.validated = !0);
          else if (a) {
            var X = p(a);
            if (typeof X == "function" && X !== a.entries)
              for (var J = X.call(a), H; !(H = J.next()).done; )
                or(H.value) && $r(H.value, h);
          }
        }
      }
      function jt(a) {
        {
          var h = a.type;
          if (h == null || typeof h == "string")
            return;
          var b;
          if (typeof h == "function")
            b = h.propTypes;
          else if (typeof h == "object" && (h.$$typeof === d || // Note: Memo only checks outer props here.
          // Inner props are checked in the reconciler.
          h.$$typeof === v))
            b = h.propTypes;
          else
            return;
          if (b) {
            var D = Q(h);
            w(b, a.props, "prop", D, a);
          } else if (h.PropTypes !== void 0 && !nr) {
            nr = !0;
            var X = Q(h);
            O("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", X || "Unknown");
          }
          typeof h.getDefaultProps == "function" && !h.getDefaultProps.isReactClassApproved && O("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
        }
      }
      function Lt(a) {
        {
          for (var h = Object.keys(a.props), b = 0; b < h.length; b++) {
            var D = h[b];
            if (D !== "children" && D !== "key") {
              be(a), O("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", D), be(null);
              break;
            }
          }
          a.ref !== null && (be(a), O("Invalid attribute `ref` supplied to `React.Fragment`."), be(null));
        }
      }
      function Dr(a, h, b, D, X, J) {
        {
          var H = ve(a);
          if (!H) {
            var I = "";
            (a === void 0 || typeof a == "object" && a !== null && Object.keys(a).length === 0) && (I += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
            var oe = Dt();
            oe ? I += oe : I += Ar();
            var me;
            a === null ? me = "null" : Array.isArray(a) ? me = "array" : a !== void 0 && a.$$typeof === t ? (me = "<" + (Q(a.type) || "Unknown") + " />", I = " Did you accidentally export a JSX literal instead of a component?") : me = typeof a, O("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", me, I);
          }
          var Re = Le(a, h, b, X, J);
          if (Re == null)
            return Re;
          if (H) {
            var Te = h.children;
            if (Te !== void 0)
              if (D)
                if (Array.isArray(Te)) {
                  for (var ir = 0; ir < Te.length; ir++)
                    Nr(Te[ir], a);
                  Object.freeze && Object.freeze(Te);
                } else
                  O("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
              else
                Nr(Te, a);
          }
          return a === e.Fragment ? Lt(Re) : jt(Re), Re;
        }
      }
      function Mt(a, h, b) {
        return Dr(a, h, b, !0);
      }
      function Ft(a, h, b) {
        return Dr(a, h, b, !1);
      }
      var qt = Ft, Vt = Mt;
      e.jsx = qt, e.jsxs = Vt;
    }();
  }(ar)), ar;
}
process.env.NODE_ENV === "production" ? hr.exports = Kt() : hr.exports = Gt();
var j = hr.exports;
function vr() {
  return vr = Object.assign ? Object.assign.bind() : function(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r];
      for (var n in t)
        Object.prototype.hasOwnProperty.call(t, n) && (e[n] = t[n]);
    }
    return e;
  }, vr.apply(this, arguments);
}
function Jt(e) {
  var r = /* @__PURE__ */ Object.create(null);
  return function(t) {
    return r[t] === void 0 && (r[t] = e(t)), r[t];
  };
}
var Xt = /^((children|dangerouslySetInnerHTML|key|ref|autoFocus|defaultValue|defaultChecked|innerHTML|suppressContentEditableWarning|suppressHydrationWarning|valueLink|abbr|accept|acceptCharset|accessKey|action|allow|allowUserMedia|allowPaymentRequest|allowFullScreen|allowTransparency|alt|async|autoComplete|autoPlay|capture|cellPadding|cellSpacing|challenge|charSet|checked|cite|classID|className|cols|colSpan|content|contentEditable|contextMenu|controls|controlsList|coords|crossOrigin|data|dateTime|decoding|default|defer|dir|disabled|disablePictureInPicture|download|draggable|encType|enterKeyHint|form|formAction|formEncType|formMethod|formNoValidate|formTarget|frameBorder|headers|height|hidden|high|href|hrefLang|htmlFor|httpEquiv|id|inputMode|integrity|is|keyParams|keyType|kind|label|lang|list|loading|loop|low|marginHeight|marginWidth|max|maxLength|media|mediaGroup|method|min|minLength|multiple|muted|name|nonce|noValidate|open|optimum|pattern|placeholder|playsInline|poster|preload|profile|radioGroup|readOnly|referrerPolicy|rel|required|reversed|role|rows|rowSpan|sandbox|scope|scoped|scrolling|seamless|selected|shape|size|sizes|slot|span|spellCheck|src|srcDoc|srcLang|srcSet|start|step|style|summary|tabIndex|target|title|translate|type|useMap|value|width|wmode|wrap|about|datatype|inlist|prefix|property|resource|typeof|vocab|autoCapitalize|autoCorrect|autoSave|color|incremental|fallback|inert|itemProp|itemScope|itemType|itemID|itemRef|on|option|results|security|unselectable|accentHeight|accumulate|additive|alignmentBaseline|allowReorder|alphabetic|amplitude|arabicForm|ascent|attributeName|attributeType|autoReverse|azimuth|baseFrequency|baselineShift|baseProfile|bbox|begin|bias|by|calcMode|capHeight|clip|clipPathUnits|clipPath|clipRule|colorInterpolation|colorInterpolationFilters|colorProfile|colorRendering|contentScriptType|contentStyleType|cursor|cx|cy|d|decelerate|descent|diffuseConstant|direction|display|divisor|dominantBaseline|dur|dx|dy|edgeMode|elevation|enableBackground|end|exponent|externalResourcesRequired|fill|fillOpacity|fillRule|filter|filterRes|filterUnits|floodColor|floodOpacity|focusable|fontFamily|fontSize|fontSizeAdjust|fontStretch|fontStyle|fontVariant|fontWeight|format|from|fr|fx|fy|g1|g2|glyphName|glyphOrientationHorizontal|glyphOrientationVertical|glyphRef|gradientTransform|gradientUnits|hanging|horizAdvX|horizOriginX|ideographic|imageRendering|in|in2|intercept|k|k1|k2|k3|k4|kernelMatrix|kernelUnitLength|kerning|keyPoints|keySplines|keyTimes|lengthAdjust|letterSpacing|lightingColor|limitingConeAngle|local|markerEnd|markerMid|markerStart|markerHeight|markerUnits|markerWidth|mask|maskContentUnits|maskUnits|mathematical|mode|numOctaves|offset|opacity|operator|order|orient|orientation|origin|overflow|overlinePosition|overlineThickness|panose1|paintOrder|pathLength|patternContentUnits|patternTransform|patternUnits|pointerEvents|points|pointsAtX|pointsAtY|pointsAtZ|preserveAlpha|preserveAspectRatio|primitiveUnits|r|radius|refX|refY|renderingIntent|repeatCount|repeatDur|requiredExtensions|requiredFeatures|restart|result|rotate|rx|ry|scale|seed|shapeRendering|slope|spacing|specularConstant|specularExponent|speed|spreadMethod|startOffset|stdDeviation|stemh|stemv|stitchTiles|stopColor|stopOpacity|strikethroughPosition|strikethroughThickness|string|stroke|strokeDasharray|strokeDashoffset|strokeLinecap|strokeLinejoin|strokeMiterlimit|strokeOpacity|strokeWidth|surfaceScale|systemLanguage|tableValues|targetX|targetY|textAnchor|textDecoration|textRendering|textLength|to|transform|u1|u2|underlinePosition|underlineThickness|unicode|unicodeBidi|unicodeRange|unitsPerEm|vAlphabetic|vHanging|vIdeographic|vMathematical|values|vectorEffect|version|vertAdvY|vertOriginX|vertOriginY|viewBox|viewTarget|visibility|widths|wordSpacing|writingMode|x|xHeight|x1|x2|xChannelSelector|xlinkActuate|xlinkArcrole|xlinkHref|xlinkRole|xlinkShow|xlinkTitle|xlinkType|xmlBase|xmlns|xmlnsXlink|xmlLang|xmlSpace|y|y1|y2|yChannelSelector|z|zoomAndPan|for|class|autofocus)|(([Dd][Aa][Tt][Aa]|[Aa][Rr][Ii][Aa]|x)-.*))$/, Zt = /* @__PURE__ */ Jt(
  function(e) {
    return Xt.test(e) || e.charCodeAt(0) === 111 && e.charCodeAt(1) === 110 && e.charCodeAt(2) < 91;
  }
  /* Z+1 */
);
function Qt(e) {
  if (e.sheet)
    return e.sheet;
  for (var r = 0; r < document.styleSheets.length; r++)
    if (document.styleSheets[r].ownerNode === e)
      return document.styleSheets[r];
}
function en(e) {
  var r = document.createElement("style");
  return r.setAttribute("data-emotion", e.key), e.nonce !== void 0 && r.setAttribute("nonce", e.nonce), r.appendChild(document.createTextNode("")), r.setAttribute("data-s", ""), r;
}
var rn = /* @__PURE__ */ function() {
  function e(t) {
    var n = this;
    this._insertTag = function(i) {
      var o;
      n.tags.length === 0 ? n.insertionPoint ? o = n.insertionPoint.nextSibling : n.prepend ? o = n.container.firstChild : o = n.before : o = n.tags[n.tags.length - 1].nextSibling, n.container.insertBefore(i, o), n.tags.push(i);
    }, this.isSpeedy = t.speedy === void 0 ? process.env.NODE_ENV === "production" : t.speedy, this.tags = [], this.ctr = 0, this.nonce = t.nonce, this.key = t.key, this.container = t.container, this.prepend = t.prepend, this.insertionPoint = t.insertionPoint, this.before = null;
  }
  var r = e.prototype;
  return r.hydrate = function(n) {
    n.forEach(this._insertTag);
  }, r.insert = function(n) {
    this.ctr % (this.isSpeedy ? 65e3 : 1) === 0 && this._insertTag(en(this));
    var i = this.tags[this.tags.length - 1];
    if (process.env.NODE_ENV !== "production") {
      var o = n.charCodeAt(0) === 64 && n.charCodeAt(1) === 105;
      o && this._alreadyInsertedOrderInsensitiveRule && console.error(`You're attempting to insert the following rule:
` + n + "\n\n`@import` rules must be before all other types of rules in a stylesheet but other rules have already been inserted. Please ensure that `@import` rules are before all other rules."), this._alreadyInsertedOrderInsensitiveRule = this._alreadyInsertedOrderInsensitiveRule || !o;
    }
    if (this.isSpeedy) {
      var s = Qt(i);
      try {
        s.insertRule(n, s.cssRules.length);
      } catch (c) {
        process.env.NODE_ENV !== "production" && !/:(-moz-placeholder|-moz-focus-inner|-moz-focusring|-ms-input-placeholder|-moz-read-write|-moz-read-only|-ms-clear|-ms-expand|-ms-reveal){/.test(n) && console.error('There was a problem inserting the following rule: "' + n + '"', c);
      }
    } else
      i.appendChild(document.createTextNode(n));
    this.ctr++;
  }, r.flush = function() {
    this.tags.forEach(function(n) {
      return n.parentNode && n.parentNode.removeChild(n);
    }), this.tags = [], this.ctr = 0, process.env.NODE_ENV !== "production" && (this._alreadyInsertedOrderInsensitiveRule = !1);
  }, e;
}(), te = "-ms-", Je = "-moz-", q = "-webkit-", Er = "comm", xr = "rule", Sr = "decl", tn = "@import", ut = "@keyframes", nn = "@layer", on = Math.abs, er = String.fromCharCode, an = Object.assign;
function sn(e, r) {
  return ee(e, 0) ^ 45 ? (((r << 2 ^ ee(e, 0)) << 2 ^ ee(e, 1)) << 2 ^ ee(e, 2)) << 2 ^ ee(e, 3) : 0;
}
function dt(e) {
  return e.trim();
}
function cn(e, r) {
  return (e = r.exec(e)) ? e[0] : e;
}
function V(e, r, t) {
  return e.replace(r, t);
}
function mr(e, r) {
  return e.indexOf(r);
}
function ee(e, r) {
  return e.charCodeAt(r) | 0;
}
function qe(e, r, t) {
  return e.slice(r, t);
}
function ye(e) {
  return e.length;
}
function Rr(e) {
  return e.length;
}
function We(e, r) {
  return r.push(e), e;
}
function ln(e, r) {
  return e.map(r).join("");
}
var rr = 1, $e = 1, ft = 0, ae = 0, Z = 0, De = "";
function tr(e, r, t, n, i, o, s) {
  return { value: e, root: r, parent: t, type: n, props: i, children: o, line: rr, column: $e, length: s, return: "" };
}
function Me(e, r) {
  return an(tr("", null, null, "", null, null, 0), e, { length: -e.length }, r);
}
function un() {
  return Z;
}
function dn() {
  return Z = ae > 0 ? ee(De, --ae) : 0, $e--, Z === 10 && ($e = 1, rr--), Z;
}
function ce() {
  return Z = ae < ft ? ee(De, ae++) : 0, $e++, Z === 10 && ($e = 1, rr++), Z;
}
function ge() {
  return ee(De, ae);
}
function He() {
  return ae;
}
function ze(e, r) {
  return qe(De, e, r);
}
function Ve(e) {
  switch (e) {
    case 0:
    case 9:
    case 10:
    case 13:
    case 32:
      return 5;
    case 33:
    case 43:
    case 44:
    case 47:
    case 62:
    case 64:
    case 126:
    case 59:
    case 123:
    case 125:
      return 4;
    case 58:
      return 3;
    case 34:
    case 39:
    case 40:
    case 91:
      return 2;
    case 41:
    case 93:
      return 1;
  }
  return 0;
}
function pt(e) {
  return rr = $e = 1, ft = ye(De = e), ae = 0, [];
}
function ht(e) {
  return De = "", e;
}
function Ke(e) {
  return dt(ze(ae - 1, yr(e === 91 ? e + 2 : e === 40 ? e + 1 : e)));
}
function fn(e) {
  for (; (Z = ge()) && Z < 33; )
    ce();
  return Ve(e) > 2 || Ve(Z) > 3 ? "" : " ";
}
function pn(e, r) {
  for (; --r && ce() && !(Z < 48 || Z > 102 || Z > 57 && Z < 65 || Z > 70 && Z < 97); )
    ;
  return ze(e, He() + (r < 6 && ge() == 32 && ce() == 32));
}
function yr(e) {
  for (; ce(); )
    switch (Z) {
      case e:
        return ae;
      case 34:
      case 39:
        e !== 34 && e !== 39 && yr(Z);
        break;
      case 40:
        e === 41 && yr(e);
        break;
      case 92:
        ce();
        break;
    }
  return ae;
}
function hn(e, r) {
  for (; ce() && e + Z !== 57; )
    if (e + Z === 84 && ge() === 47)
      break;
  return "/*" + ze(r, ae - 1) + "*" + er(e === 47 ? e : ce());
}
function vn(e) {
  for (; !Ve(ge()); )
    ce();
  return ze(e, ae);
}
function mn(e) {
  return ht(Ge("", null, null, null, [""], e = pt(e), 0, [0], e));
}
function Ge(e, r, t, n, i, o, s, c, d) {
  for (var y = 0, u = 0, v = s, E = 0, _ = 0, S = 0, m = 1, x = 1, T = 1, R = 0, N = "", W = i, p = o, K = n, O = N; x; )
    switch (S = R, R = ce()) {
      case 40:
        if (S != 108 && ee(O, v - 1) == 58) {
          mr(O += V(Ke(R), "&", "&\f"), "&\f") != -1 && (T = -1);
          break;
        }
      case 34:
      case 39:
      case 91:
        O += Ke(R);
        break;
      case 9:
      case 10:
      case 13:
      case 32:
        O += fn(S);
        break;
      case 92:
        O += pn(He() - 1, 7);
        continue;
      case 47:
        switch (ge()) {
          case 42:
          case 47:
            We(yn(hn(ce(), He()), r, t), d);
            break;
          default:
            O += "/";
        }
        break;
      case 123 * m:
        c[y++] = ye(O) * T;
      case 125 * m:
      case 59:
      case 0:
        switch (R) {
          case 0:
          case 125:
            x = 0;
          case 59 + u:
            T == -1 && (O = V(O, /\f/g, "")), _ > 0 && ye(O) - v && We(_ > 32 ? Mr(O + ";", n, t, v - 1) : Mr(V(O, " ", "") + ";", n, t, v - 2), d);
            break;
          case 59:
            O += ";";
          default:
            if (We(K = Lr(O, r, t, y, u, i, c, N, W = [], p = [], v), o), R === 123)
              if (u === 0)
                Ge(O, r, K, K, W, o, v, c, p);
              else
                switch (E === 99 && ee(O, 3) === 110 ? 100 : E) {
                  case 100:
                  case 108:
                  case 109:
                  case 115:
                    Ge(e, K, K, n && We(Lr(e, K, K, 0, 0, i, c, N, i, W = [], v), p), i, p, v, c, n ? W : p);
                    break;
                  default:
                    Ge(O, K, K, K, [""], p, 0, c, p);
                }
        }
        y = u = _ = 0, m = T = 1, N = O = "", v = s;
        break;
      case 58:
        v = 1 + ye(O), _ = S;
      default:
        if (m < 1) {
          if (R == 123)
            --m;
          else if (R == 125 && m++ == 0 && dn() == 125)
            continue;
        }
        switch (O += er(R), R * m) {
          case 38:
            T = u > 0 ? 1 : (O += "\f", -1);
            break;
          case 44:
            c[y++] = (ye(O) - 1) * T, T = 1;
            break;
          case 64:
            ge() === 45 && (O += Ke(ce())), E = ge(), u = v = ye(N = O += vn(He())), R++;
            break;
          case 45:
            S === 45 && ye(O) == 2 && (m = 0);
        }
    }
  return o;
}
function Lr(e, r, t, n, i, o, s, c, d, y, u) {
  for (var v = i - 1, E = i === 0 ? o : [""], _ = Rr(E), S = 0, m = 0, x = 0; S < n; ++S)
    for (var T = 0, R = qe(e, v + 1, v = on(m = s[S])), N = e; T < _; ++T)
      (N = dt(m > 0 ? E[T] + " " + R : V(R, /&\f/g, E[T]))) && (d[x++] = N);
  return tr(e, r, t, i === 0 ? xr : c, d, y, u);
}
function yn(e, r, t) {
  return tr(e, r, t, Er, er(un()), qe(e, 2, -2), 0);
}
function Mr(e, r, t, n) {
  return tr(e, r, t, Sr, qe(e, 0, n), qe(e, n + 1, -1), n);
}
function ke(e, r) {
  for (var t = "", n = Rr(e), i = 0; i < n; i++)
    t += r(e[i], i, e, r) || "";
  return t;
}
function gn(e, r, t, n) {
  switch (e.type) {
    case nn:
      if (e.children.length) break;
    case tn:
    case Sr:
      return e.return = e.return || e.value;
    case Er:
      return "";
    case ut:
      return e.return = e.value + "{" + ke(e.children, n) + "}";
    case xr:
      e.value = e.props.join(",");
  }
  return ye(t = ke(e.children, n)) ? e.return = e.value + "{" + t + "}" : "";
}
function bn(e) {
  var r = Rr(e);
  return function(t, n, i, o) {
    for (var s = "", c = 0; c < r; c++)
      s += e[c](t, n, i, o) || "";
    return s;
  };
}
function wn(e) {
  return function(r) {
    r.root || (r = r.return) && e(r);
  };
}
var En = function(r, t, n) {
  for (var i = 0, o = 0; i = o, o = ge(), i === 38 && o === 12 && (t[n] = 1), !Ve(o); )
    ce();
  return ze(r, ae);
}, xn = function(r, t) {
  var n = -1, i = 44;
  do
    switch (Ve(i)) {
      case 0:
        i === 38 && ge() === 12 && (t[n] = 1), r[n] += En(ae - 1, t, n);
        break;
      case 2:
        r[n] += Ke(i);
        break;
      case 4:
        if (i === 44) {
          r[++n] = ge() === 58 ? "&\f" : "", t[n] = r[n].length;
          break;
        }
      default:
        r[n] += er(i);
    }
  while (i = ce());
  return r;
}, Sn = function(r, t) {
  return ht(xn(pt(r), t));
}, Fr = /* @__PURE__ */ new WeakMap(), Rn = function(r) {
  if (!(r.type !== "rule" || !r.parent || // positive .length indicates that this rule contains pseudo
  // negative .length indicates that this rule has been already prefixed
  r.length < 1)) {
    for (var t = r.value, n = r.parent, i = r.column === n.column && r.line === n.line; n.type !== "rule"; )
      if (n = n.parent, !n) return;
    if (!(r.props.length === 1 && t.charCodeAt(0) !== 58 && !Fr.get(n)) && !i) {
      Fr.set(r, !0);
      for (var o = [], s = Sn(t, o), c = n.props, d = 0, y = 0; d < s.length; d++)
        for (var u = 0; u < c.length; u++, y++)
          r.props[y] = o[d] ? s[d].replace(/&\f/g, c[u]) : c[u] + " " + s[d];
    }
  }
}, On = function(r) {
  if (r.type === "decl") {
    var t = r.value;
    // charcode for l
    t.charCodeAt(0) === 108 && // charcode for b
    t.charCodeAt(2) === 98 && (r.return = "", r.value = "");
  }
}, _n = "emotion-disable-server-rendering-unsafe-selector-warning-please-do-not-use-this-the-warning-exists-for-a-reason", Cn = function(r) {
  return r.type === "comm" && r.children.indexOf(_n) > -1;
}, Tn = function(r) {
  return function(t, n, i) {
    if (!(t.type !== "rule" || r.compat)) {
      var o = t.value.match(/(:first|:nth|:nth-last)-child/g);
      if (o) {
        for (var s = !!t.parent, c = s ? t.parent.children : (
          // global rule at the root level
          i
        ), d = c.length - 1; d >= 0; d--) {
          var y = c[d];
          if (y.line < t.line)
            break;
          if (y.column < t.column) {
            if (Cn(y))
              return;
            break;
          }
        }
        o.forEach(function(u) {
          console.error('The pseudo class "' + u + '" is potentially unsafe when doing server-side rendering. Try changing it to "' + u.split("-child")[0] + '-of-type".');
        });
      }
    }
  };
}, vt = function(r) {
  return r.type.charCodeAt(1) === 105 && r.type.charCodeAt(0) === 64;
}, Pn = function(r, t) {
  for (var n = r - 1; n >= 0; n--)
    if (!vt(t[n]))
      return !0;
  return !1;
}, qr = function(r) {
  r.type = "", r.value = "", r.return = "", r.children = "", r.props = "";
}, An = function(r, t, n) {
  vt(r) && (r.parent ? (console.error("`@import` rules can't be nested inside other rules. Please move it to the top level and put it before regular rules. Keep in mind that they can only be used within global styles."), qr(r)) : Pn(t, n) && (console.error("`@import` rules can't be after other rules. Please put your `@import` rules before your other rules."), qr(r)));
};
function mt(e, r) {
  switch (sn(e, r)) {
    case 5103:
      return q + "print-" + e + e;
    case 5737:
    case 4201:
    case 3177:
    case 3433:
    case 1641:
    case 4457:
    case 2921:
    case 5572:
    case 6356:
    case 5844:
    case 3191:
    case 6645:
    case 3005:
    case 6391:
    case 5879:
    case 5623:
    case 6135:
    case 4599:
    case 4855:
    case 4215:
    case 6389:
    case 5109:
    case 5365:
    case 5621:
    case 3829:
      return q + e + e;
    case 5349:
    case 4246:
    case 4810:
    case 6968:
    case 2756:
      return q + e + Je + e + te + e + e;
    case 6828:
    case 4268:
      return q + e + te + e + e;
    case 6165:
      return q + e + te + "flex-" + e + e;
    case 5187:
      return q + e + V(e, /(\w+).+(:[^]+)/, q + "box-$1$2" + te + "flex-$1$2") + e;
    case 5443:
      return q + e + te + "flex-item-" + V(e, /flex-|-self/, "") + e;
    case 4675:
      return q + e + te + "flex-line-pack" + V(e, /align-content|flex-|-self/, "") + e;
    case 5548:
      return q + e + te + V(e, "shrink", "negative") + e;
    case 5292:
      return q + e + te + V(e, "basis", "preferred-size") + e;
    case 6060:
      return q + "box-" + V(e, "-grow", "") + q + e + te + V(e, "grow", "positive") + e;
    case 4554:
      return q + V(e, /([^-])(transform)/g, "$1" + q + "$2") + e;
    case 6187:
      return V(V(V(e, /(zoom-|grab)/, q + "$1"), /(image-set)/, q + "$1"), e, "") + e;
    case 5495:
    case 3959:
      return V(e, /(image-set\([^]*)/, q + "$1$`$1");
    case 4968:
      return V(V(e, /(.+:)(flex-)?(.*)/, q + "box-pack:$3" + te + "flex-pack:$3"), /s.+-b[^;]+/, "justify") + q + e + e;
    case 4095:
    case 3583:
    case 4068:
    case 2532:
      return V(e, /(.+)-inline(.+)/, q + "$1$2") + e;
    case 8116:
    case 7059:
    case 5753:
    case 5535:
    case 5445:
    case 5701:
    case 4933:
    case 4677:
    case 5533:
    case 5789:
    case 5021:
    case 4765:
      if (ye(e) - 1 - r > 6) switch (ee(e, r + 1)) {
        case 109:
          if (ee(e, r + 4) !== 45) break;
        case 102:
          return V(e, /(.+:)(.+)-([^]+)/, "$1" + q + "$2-$3$1" + Je + (ee(e, r + 3) == 108 ? "$3" : "$2-$3")) + e;
        case 115:
          return ~mr(e, "stretch") ? mt(V(e, "stretch", "fill-available"), r) + e : e;
      }
      break;
    case 4949:
      if (ee(e, r + 1) !== 115) break;
    case 6444:
      switch (ee(e, ye(e) - 3 - (~mr(e, "!important") && 10))) {
        case 107:
          return V(e, ":", ":" + q) + e;
        case 101:
          return V(e, /(.+:)([^;!]+)(;|!.+)?/, "$1" + q + (ee(e, 14) === 45 ? "inline-" : "") + "box$3$1" + q + "$2$3$1" + te + "$2box$3") + e;
      }
      break;
    case 5936:
      switch (ee(e, r + 11)) {
        case 114:
          return q + e + te + V(e, /[svh]\w+-[tblr]{2}/, "tb") + e;
        case 108:
          return q + e + te + V(e, /[svh]\w+-[tblr]{2}/, "tb-rl") + e;
        case 45:
          return q + e + te + V(e, /[svh]\w+-[tblr]{2}/, "lr") + e;
      }
      return q + e + te + e + e;
  }
  return e;
}
var kn = function(r, t, n, i) {
  if (r.length > -1 && !r.return) switch (r.type) {
    case Sr:
      r.return = mt(r.value, r.length);
      break;
    case ut:
      return ke([Me(r, {
        value: V(r.value, "@", "@" + q)
      })], i);
    case xr:
      if (r.length) return ln(r.props, function(o) {
        switch (cn(o, /(::plac\w+|:read-\w+)/)) {
          case ":read-only":
          case ":read-write":
            return ke([Me(r, {
              props: [V(o, /:(read-\w+)/, ":" + Je + "$1")]
            })], i);
          case "::placeholder":
            return ke([Me(r, {
              props: [V(o, /:(plac\w+)/, ":" + q + "input-$1")]
            }), Me(r, {
              props: [V(o, /:(plac\w+)/, ":" + Je + "$1")]
            }), Me(r, {
              props: [V(o, /:(plac\w+)/, te + "input-$1")]
            })], i);
        }
        return "";
      });
  }
}, $n = [kn], Nn = function(r) {
  var t = r.key;
  if (process.env.NODE_ENV !== "production" && !t)
    throw new Error(`You have to configure \`key\` for your cache. Please make sure it's unique (and not equal to 'css') as it's used for linking styles to your cache.
If multiple caches share the same key they might "fight" for each other's style elements.`);
  if (t === "css") {
    var n = document.querySelectorAll("style[data-emotion]:not([data-s])");
    Array.prototype.forEach.call(n, function(m) {
      var x = m.getAttribute("data-emotion");
      x.indexOf(" ") !== -1 && (document.head.appendChild(m), m.setAttribute("data-s", ""));
    });
  }
  var i = r.stylisPlugins || $n;
  if (process.env.NODE_ENV !== "production" && /[^a-z-]/.test(t))
    throw new Error('Emotion key must only contain lower case alphabetical characters and - but "' + t + '" was passed');
  var o = {}, s, c = [];
  s = r.container || document.head, Array.prototype.forEach.call(
    // this means we will ignore elements which don't have a space in them which
    // means that the style elements we're looking at are only Emotion 11 server-rendered style elements
    document.querySelectorAll('style[data-emotion^="' + t + ' "]'),
    function(m) {
      for (var x = m.getAttribute("data-emotion").split(" "), T = 1; T < x.length; T++)
        o[x[T]] = !0;
      c.push(m);
    }
  );
  var d, y = [Rn, On];
  process.env.NODE_ENV !== "production" && y.push(Tn({
    get compat() {
      return S.compat;
    }
  }), An);
  {
    var u, v = [gn, process.env.NODE_ENV !== "production" ? function(m) {
      m.root || (m.return ? u.insert(m.return) : m.value && m.type !== Er && u.insert(m.value + "{}"));
    } : wn(function(m) {
      u.insert(m);
    })], E = bn(y.concat(i, v)), _ = function(x) {
      return ke(mn(x), E);
    };
    d = function(x, T, R, N) {
      u = R, process.env.NODE_ENV !== "production" && T.map !== void 0 && (u = {
        insert: function(p) {
          R.insert(p + T.map);
        }
      }), _(x ? x + "{" + T.styles + "}" : T.styles), N && (S.inserted[T.name] = !0);
    };
  }
  var S = {
    key: t,
    sheet: new rn({
      key: t,
      container: s,
      nonce: r.nonce,
      speedy: r.speedy,
      prepend: r.prepend,
      insertionPoint: r.insertionPoint
    }),
    nonce: r.nonce,
    inserted: o,
    registered: {},
    insert: d
  };
  return S.sheet.hydrate(c), S;
}, gr = { exports: {} }, B = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Vr;
function Dn() {
  if (Vr) return B;
  Vr = 1;
  var e = typeof Symbol == "function" && Symbol.for, r = e ? Symbol.for("react.element") : 60103, t = e ? Symbol.for("react.portal") : 60106, n = e ? Symbol.for("react.fragment") : 60107, i = e ? Symbol.for("react.strict_mode") : 60108, o = e ? Symbol.for("react.profiler") : 60114, s = e ? Symbol.for("react.provider") : 60109, c = e ? Symbol.for("react.context") : 60110, d = e ? Symbol.for("react.async_mode") : 60111, y = e ? Symbol.for("react.concurrent_mode") : 60111, u = e ? Symbol.for("react.forward_ref") : 60112, v = e ? Symbol.for("react.suspense") : 60113, E = e ? Symbol.for("react.suspense_list") : 60120, _ = e ? Symbol.for("react.memo") : 60115, S = e ? Symbol.for("react.lazy") : 60116, m = e ? Symbol.for("react.block") : 60121, x = e ? Symbol.for("react.fundamental") : 60117, T = e ? Symbol.for("react.responder") : 60118, R = e ? Symbol.for("react.scope") : 60119;
  function N(p) {
    if (typeof p == "object" && p !== null) {
      var K = p.$$typeof;
      switch (K) {
        case r:
          switch (p = p.type, p) {
            case d:
            case y:
            case n:
            case o:
            case i:
            case v:
              return p;
            default:
              switch (p = p && p.$$typeof, p) {
                case c:
                case u:
                case S:
                case _:
                case s:
                  return p;
                default:
                  return K;
              }
          }
        case t:
          return K;
      }
    }
  }
  function W(p) {
    return N(p) === y;
  }
  return B.AsyncMode = d, B.ConcurrentMode = y, B.ContextConsumer = c, B.ContextProvider = s, B.Element = r, B.ForwardRef = u, B.Fragment = n, B.Lazy = S, B.Memo = _, B.Portal = t, B.Profiler = o, B.StrictMode = i, B.Suspense = v, B.isAsyncMode = function(p) {
    return W(p) || N(p) === d;
  }, B.isConcurrentMode = W, B.isContextConsumer = function(p) {
    return N(p) === c;
  }, B.isContextProvider = function(p) {
    return N(p) === s;
  }, B.isElement = function(p) {
    return typeof p == "object" && p !== null && p.$$typeof === r;
  }, B.isForwardRef = function(p) {
    return N(p) === u;
  }, B.isFragment = function(p) {
    return N(p) === n;
  }, B.isLazy = function(p) {
    return N(p) === S;
  }, B.isMemo = function(p) {
    return N(p) === _;
  }, B.isPortal = function(p) {
    return N(p) === t;
  }, B.isProfiler = function(p) {
    return N(p) === o;
  }, B.isStrictMode = function(p) {
    return N(p) === i;
  }, B.isSuspense = function(p) {
    return N(p) === v;
  }, B.isValidElementType = function(p) {
    return typeof p == "string" || typeof p == "function" || p === n || p === y || p === o || p === i || p === v || p === E || typeof p == "object" && p !== null && (p.$$typeof === S || p.$$typeof === _ || p.$$typeof === s || p.$$typeof === c || p.$$typeof === u || p.$$typeof === x || p.$$typeof === T || p.$$typeof === R || p.$$typeof === m);
  }, B.typeOf = N, B;
}
var z = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Yr;
function In() {
  return Yr || (Yr = 1, process.env.NODE_ENV !== "production" && function() {
    var e = typeof Symbol == "function" && Symbol.for, r = e ? Symbol.for("react.element") : 60103, t = e ? Symbol.for("react.portal") : 60106, n = e ? Symbol.for("react.fragment") : 60107, i = e ? Symbol.for("react.strict_mode") : 60108, o = e ? Symbol.for("react.profiler") : 60114, s = e ? Symbol.for("react.provider") : 60109, c = e ? Symbol.for("react.context") : 60110, d = e ? Symbol.for("react.async_mode") : 60111, y = e ? Symbol.for("react.concurrent_mode") : 60111, u = e ? Symbol.for("react.forward_ref") : 60112, v = e ? Symbol.for("react.suspense") : 60113, E = e ? Symbol.for("react.suspense_list") : 60120, _ = e ? Symbol.for("react.memo") : 60115, S = e ? Symbol.for("react.lazy") : 60116, m = e ? Symbol.for("react.block") : 60121, x = e ? Symbol.for("react.fundamental") : 60117, T = e ? Symbol.for("react.responder") : 60118, R = e ? Symbol.for("react.scope") : 60119;
    function N(g) {
      return typeof g == "string" || typeof g == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      g === n || g === y || g === o || g === i || g === v || g === E || typeof g == "object" && g !== null && (g.$$typeof === S || g.$$typeof === _ || g.$$typeof === s || g.$$typeof === c || g.$$typeof === u || g.$$typeof === x || g.$$typeof === T || g.$$typeof === R || g.$$typeof === m);
    }
    function W(g) {
      if (typeof g == "object" && g !== null) {
        var Le = g.$$typeof;
        switch (Le) {
          case r:
            var Se = g.type;
            switch (Se) {
              case d:
              case y:
              case n:
              case o:
              case i:
              case v:
                return Se;
              default:
                var be = Se && Se.$$typeof;
                switch (be) {
                  case c:
                  case u:
                  case S:
                  case _:
                  case s:
                    return be;
                  default:
                    return Le;
                }
            }
          case t:
            return Le;
        }
      }
    }
    var p = d, K = y, O = c, Oe = s, he = r, ve = u, _e = n, Ie = S, xe = _, je = t, le = o, Q = i, Ce = v, de = !1;
    function f(g) {
      return de || (de = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), w(g) || W(g) === d;
    }
    function w(g) {
      return W(g) === y;
    }
    function P(g) {
      return W(g) === c;
    }
    function A(g) {
      return W(g) === s;
    }
    function k(g) {
      return typeof g == "object" && g !== null && g.$$typeof === r;
    }
    function M(g) {
      return W(g) === u;
    }
    function $(g) {
      return W(g) === n;
    }
    function C(g) {
      return W(g) === S;
    }
    function L(g) {
      return W(g) === _;
    }
    function Y(g) {
      return W(g) === t;
    }
    function U(g) {
      return W(g) === o;
    }
    function ue(g) {
      return W(g) === i;
    }
    function se(g) {
      return W(g) === v;
    }
    z.AsyncMode = p, z.ConcurrentMode = K, z.ContextConsumer = O, z.ContextProvider = Oe, z.Element = he, z.ForwardRef = ve, z.Fragment = _e, z.Lazy = Ie, z.Memo = xe, z.Portal = je, z.Profiler = le, z.StrictMode = Q, z.Suspense = Ce, z.isAsyncMode = f, z.isConcurrentMode = w, z.isContextConsumer = P, z.isContextProvider = A, z.isElement = k, z.isForwardRef = M, z.isFragment = $, z.isLazy = C, z.isMemo = L, z.isPortal = Y, z.isProfiler = U, z.isStrictMode = ue, z.isSuspense = se, z.isValidElementType = N, z.typeOf = W;
  }()), z;
}
process.env.NODE_ENV === "production" ? gr.exports = Dn() : gr.exports = In();
var yt = gr.exports, jn = !0;
function Ln(e, r, t) {
  var n = "";
  return t.split(" ").forEach(function(i) {
    e[i] !== void 0 ? r.push(e[i] + ";") : n += i + " ";
  }), n;
}
var gt = function(r, t, n) {
  var i = r.key + "-" + t.name;
  // we only need to add the styles to the registered cache if the
  // class name could be used further down
  // the tree but if it's a string tag, we know it won't
  // so we don't have to add it to registered cache.
  // this improves memory usage since we can avoid storing the whole style string
  (n === !1 || // we need to always store it if we're in compat mode and
  // in node since emotion-server relies on whether a style is in
  // the registered cache to know whether a style is global or not
  // also, note that this check will be dead code eliminated in the browser
  jn === !1) && r.registered[i] === void 0 && (r.registered[i] = t.styles);
}, Mn = function(r, t, n) {
  gt(r, t, n);
  var i = r.key + "-" + t.name;
  if (r.inserted[t.name] === void 0) {
    var o = t;
    do
      r.insert(t === o ? "." + i : "", o, r.sheet, !0), o = o.next;
    while (o !== void 0);
  }
};
function Fn(e) {
  for (var r = 0, t, n = 0, i = e.length; i >= 4; ++n, i -= 4)
    t = e.charCodeAt(n) & 255 | (e.charCodeAt(++n) & 255) << 8 | (e.charCodeAt(++n) & 255) << 16 | (e.charCodeAt(++n) & 255) << 24, t = /* Math.imul(k, m): */
    (t & 65535) * 1540483477 + ((t >>> 16) * 59797 << 16), t ^= /* k >>> r: */
    t >>> 24, r = /* Math.imul(k, m): */
    (t & 65535) * 1540483477 + ((t >>> 16) * 59797 << 16) ^ /* Math.imul(h, m): */
    (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16);
  switch (i) {
    case 3:
      r ^= (e.charCodeAt(n + 2) & 255) << 16;
    case 2:
      r ^= (e.charCodeAt(n + 1) & 255) << 8;
    case 1:
      r ^= e.charCodeAt(n) & 255, r = /* Math.imul(h, m): */
      (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16);
  }
  return r ^= r >>> 13, r = /* Math.imul(h, m): */
  (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16), ((r ^ r >>> 15) >>> 0).toString(36);
}
var qn = {
  animationIterationCount: 1,
  aspectRatio: 1,
  borderImageOutset: 1,
  borderImageSlice: 1,
  borderImageWidth: 1,
  boxFlex: 1,
  boxFlexGroup: 1,
  boxOrdinalGroup: 1,
  columnCount: 1,
  columns: 1,
  flex: 1,
  flexGrow: 1,
  flexPositive: 1,
  flexShrink: 1,
  flexNegative: 1,
  flexOrder: 1,
  gridRow: 1,
  gridRowEnd: 1,
  gridRowSpan: 1,
  gridRowStart: 1,
  gridColumn: 1,
  gridColumnEnd: 1,
  gridColumnSpan: 1,
  gridColumnStart: 1,
  msGridRow: 1,
  msGridRowSpan: 1,
  msGridColumn: 1,
  msGridColumnSpan: 1,
  fontWeight: 1,
  lineHeight: 1,
  opacity: 1,
  order: 1,
  orphans: 1,
  tabSize: 1,
  widows: 1,
  zIndex: 1,
  zoom: 1,
  WebkitLineClamp: 1,
  // SVG-related properties
  fillOpacity: 1,
  floodOpacity: 1,
  stopOpacity: 1,
  strokeDasharray: 1,
  strokeDashoffset: 1,
  strokeMiterlimit: 1,
  strokeOpacity: 1,
  strokeWidth: 1
};
function Vn(e) {
  var r = /* @__PURE__ */ Object.create(null);
  return function(t) {
    return r[t] === void 0 && (r[t] = e(t)), r[t];
  };
}
var Br = `You have illegal escape sequence in your template literal, most likely inside content's property value.
Because you write your CSS inside a JavaScript string you actually have to do double escaping, so for example "content: '\\00d7';" should become "content: '\\\\00d7';".
You can read more about this here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#ES2018_revision_of_illegal_escape_sequences`, Yn = "You have passed in falsy value as style object's key (can happen when in example you pass unexported component as computed key).", Bn = /[A-Z]|^ms/g, bt = /_EMO_([^_]+?)_([^]*?)_EMO_/g, Or = function(r) {
  return r.charCodeAt(1) === 45;
}, zr = function(r) {
  return r != null && typeof r != "boolean";
}, sr = /* @__PURE__ */ Vn(function(e) {
  return Or(e) ? e : e.replace(Bn, "-$&").toLowerCase();
}), Xe = function(r, t) {
  switch (r) {
    case "animation":
    case "animationName":
      if (typeof t == "string")
        return t.replace(bt, function(n, i, o) {
          return fe = {
            name: i,
            styles: o,
            next: fe
          }, i;
        });
  }
  return qn[r] !== 1 && !Or(r) && typeof t == "number" && t !== 0 ? t + "px" : t;
};
if (process.env.NODE_ENV !== "production") {
  var zn = /(var|attr|counters?|url|element|(((repeating-)?(linear|radial))|conic)-gradient)\(|(no-)?(open|close)-quote/, Wn = ["normal", "none", "initial", "inherit", "unset"], Un = Xe, Hn = /^-ms-/, Kn = /-(.)/g, Wr = {};
  Xe = function(r, t) {
    if (r === "content" && (typeof t != "string" || Wn.indexOf(t) === -1 && !zn.test(t) && (t.charAt(0) !== t.charAt(t.length - 1) || t.charAt(0) !== '"' && t.charAt(0) !== "'")))
      throw new Error("You seem to be using a value for 'content' without quotes, try replacing it with `content: '\"" + t + "\"'`");
    var n = Un(r, t);
    return n !== "" && !Or(r) && r.indexOf("-") !== -1 && Wr[r] === void 0 && (Wr[r] = !0, console.error("Using kebab-case for css properties in objects is not supported. Did you mean " + r.replace(Hn, "ms-").replace(Kn, function(i, o) {
      return o.toUpperCase();
    }) + "?")), n;
  };
}
var wt = "Component selectors can only be used in conjunction with @emotion/babel-plugin, the swc Emotion plugin, or another Emotion-aware compiler transform.";
function Ye(e, r, t) {
  if (t == null)
    return "";
  if (t.__emotion_styles !== void 0) {
    if (process.env.NODE_ENV !== "production" && t.toString() === "NO_COMPONENT_SELECTOR")
      throw new Error(wt);
    return t;
  }
  switch (typeof t) {
    case "boolean":
      return "";
    case "object": {
      if (t.anim === 1)
        return fe = {
          name: t.name,
          styles: t.styles,
          next: fe
        }, t.name;
      if (t.styles !== void 0) {
        var n = t.next;
        if (n !== void 0)
          for (; n !== void 0; )
            fe = {
              name: n.name,
              styles: n.styles,
              next: fe
            }, n = n.next;
        var i = t.styles + ";";
        return process.env.NODE_ENV !== "production" && t.map !== void 0 && (i += t.map), i;
      }
      return Gn(e, r, t);
    }
    case "function": {
      if (e !== void 0) {
        var o = fe, s = t(e);
        return fe = o, Ye(e, r, s);
      } else process.env.NODE_ENV !== "production" && console.error("Functions that are interpolated in css calls will be stringified.\nIf you want to have a css call based on props, create a function that returns a css call like this\nlet dynamicStyle = (props) => css`color: ${props.color}`\nIt can be called directly with props or interpolated in a styled call like this\nlet SomeComponent = styled('div')`${dynamicStyle}`");
      break;
    }
    case "string":
      if (process.env.NODE_ENV !== "production") {
        var c = [], d = t.replace(bt, function(y, u, v) {
          var E = "animation" + c.length;
          return c.push("const " + E + " = keyframes`" + v.replace(/^@keyframes animation-\w+/, "") + "`"), "${" + E + "}";
        });
        c.length && console.error("`keyframes` output got interpolated into plain string, please wrap it with `css`.\n\nInstead of doing this:\n\n" + [].concat(c, ["`" + d + "`"]).join(`
`) + `

You should wrap it with \`css\` like this:

` + ("css`" + d + "`"));
      }
      break;
  }
  return t;
}
function Gn(e, r, t) {
  var n = "";
  if (Array.isArray(t))
    for (var i = 0; i < t.length; i++)
      n += Ye(e, r, t[i]) + ";";
  else
    for (var o in t) {
      var s = t[o];
      if (typeof s != "object")
        zr(s) && (n += sr(o) + ":" + Xe(o, s) + ";");
      else {
        if (o === "NO_COMPONENT_SELECTOR" && process.env.NODE_ENV !== "production")
          throw new Error(wt);
        if (Array.isArray(s) && typeof s[0] == "string" && r == null)
          for (var c = 0; c < s.length; c++)
            zr(s[c]) && (n += sr(o) + ":" + Xe(o, s[c]) + ";");
        else {
          var d = Ye(e, r, s);
          switch (o) {
            case "animation":
            case "animationName": {
              n += sr(o) + ":" + d + ";";
              break;
            }
            default:
              process.env.NODE_ENV !== "production" && o === "undefined" && console.error(Yn), n += o + "{" + d + "}";
          }
        }
      }
    }
  return n;
}
var Ur = /label:\s*([^\s;\n{]+)\s*(;|$)/g, Et;
process.env.NODE_ENV !== "production" && (Et = /\/\*#\ssourceMappingURL=data:application\/json;\S+\s+\*\//g);
var fe, Hr = function(r, t, n) {
  if (r.length === 1 && typeof r[0] == "object" && r[0] !== null && r[0].styles !== void 0)
    return r[0];
  var i = !0, o = "";
  fe = void 0;
  var s = r[0];
  s == null || s.raw === void 0 ? (i = !1, o += Ye(n, t, s)) : (process.env.NODE_ENV !== "production" && s[0] === void 0 && console.error(Br), o += s[0]);
  for (var c = 1; c < r.length; c++)
    o += Ye(n, t, r[c]), i && (process.env.NODE_ENV !== "production" && s[c] === void 0 && console.error(Br), o += s[c]);
  var d;
  process.env.NODE_ENV !== "production" && (o = o.replace(Et, function(E) {
    return d = E, "";
  })), Ur.lastIndex = 0;
  for (var y = "", u; (u = Ur.exec(o)) !== null; )
    y += "-" + // $FlowFixMe we know it's not null
    u[1];
  var v = Fn(o) + y;
  return process.env.NODE_ENV !== "production" ? {
    name: v,
    styles: o,
    map: d,
    next: fe,
    toString: function() {
      return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
    }
  } : {
    name: v,
    styles: o,
    next: fe
  };
}, Jn = function(r) {
  return r();
}, Xn = ie.useInsertionEffect ? ie.useInsertionEffect : !1, xt = Xn || Jn, Zn = {}.hasOwnProperty, _r = /* @__PURE__ */ ie.createContext(
  // we're doing this to avoid preconstruct's dead code elimination in this one case
  // because this module is primarily intended for the browser and node
  // but it's also required in react native and similar environments sometimes
  // and we could have a special build just for that
  // but this is much easier and the native packages
  // might use a different theme context in the future anyway
  typeof HTMLElement < "u" ? /* @__PURE__ */ Nn({
    key: "css"
  }) : null
);
process.env.NODE_ENV !== "production" && (_r.displayName = "EmotionCacheContext");
_r.Provider;
var St = function(r) {
  return /* @__PURE__ */ zt(function(t, n) {
    var i = Wt(_r);
    return r(t, i, n);
  });
}, Cr = /* @__PURE__ */ ie.createContext({});
process.env.NODE_ENV !== "production" && (Cr.displayName = "EmotionThemeContext");
var Kr = "__EMOTION_TYPE_PLEASE_DO_NOT_USE__", Gr = "__EMOTION_LABEL_PLEASE_DO_NOT_USE__", Qn = function(r) {
  var t = r.cache, n = r.serialized, i = r.isStringTag;
  return gt(t, n, i), xt(function() {
    return Mn(t, n, i);
  }), null;
}, eo = /* @__PURE__ */ St(function(e, r, t) {
  var n = e.css;
  typeof n == "string" && r.registered[n] !== void 0 && (n = r.registered[n]);
  var i = e[Kr], o = [n], s = "";
  typeof e.className == "string" ? s = Ln(r.registered, o, e.className) : e.className != null && (s = e.className + " ");
  var c = Hr(o, void 0, ie.useContext(Cr));
  if (process.env.NODE_ENV !== "production" && c.name.indexOf("-") === -1) {
    var d = e[Gr];
    d && (c = Hr([c, "label:" + d + ";"]));
  }
  s += r.key + "-" + c.name;
  var y = {};
  for (var u in e)
    Zn.call(e, u) && u !== "css" && u !== Kr && (process.env.NODE_ENV === "production" || u !== Gr) && (y[u] = e[u]);
  return y.ref = t, y.className = s, /* @__PURE__ */ ie.createElement(ie.Fragment, null, /* @__PURE__ */ ie.createElement(Qn, {
    cache: r,
    serialized: c,
    isStringTag: typeof i == "string"
  }), /* @__PURE__ */ ie.createElement(i, y));
});
process.env.NODE_ENV !== "production" && (eo.displayName = "EmotionCssPropInternal");
var ro = !0;
function to(e, r, t) {
  var n = "";
  return t.split(" ").forEach(function(i) {
    e[i] !== void 0 ? r.push(e[i] + ";") : n += i + " ";
  }), n;
}
var Rt = function(r, t, n) {
  var i = r.key + "-" + t.name;
  // we only need to add the styles to the registered cache if the
  // class name could be used further down
  // the tree but if it's a string tag, we know it won't
  // so we don't have to add it to registered cache.
  // this improves memory usage since we can avoid storing the whole style string
  (n === !1 || // we need to always store it if we're in compat mode and
  // in node since emotion-server relies on whether a style is in
  // the registered cache to know whether a style is global or not
  // also, note that this check will be dead code eliminated in the browser
  ro === !1) && r.registered[i] === void 0 && (r.registered[i] = t.styles);
}, no = function(r, t, n) {
  Rt(r, t, n);
  var i = r.key + "-" + t.name;
  if (r.inserted[t.name] === void 0) {
    var o = t;
    do
      r.insert(t === o ? "." + i : "", o, r.sheet, !0), o = o.next;
    while (o !== void 0);
  }
};
function oo(e) {
  for (var r = 0, t, n = 0, i = e.length; i >= 4; ++n, i -= 4)
    t = e.charCodeAt(n) & 255 | (e.charCodeAt(++n) & 255) << 8 | (e.charCodeAt(++n) & 255) << 16 | (e.charCodeAt(++n) & 255) << 24, t = /* Math.imul(k, m): */
    (t & 65535) * 1540483477 + ((t >>> 16) * 59797 << 16), t ^= /* k >>> r: */
    t >>> 24, r = /* Math.imul(k, m): */
    (t & 65535) * 1540483477 + ((t >>> 16) * 59797 << 16) ^ /* Math.imul(h, m): */
    (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16);
  switch (i) {
    case 3:
      r ^= (e.charCodeAt(n + 2) & 255) << 16;
    case 2:
      r ^= (e.charCodeAt(n + 1) & 255) << 8;
    case 1:
      r ^= e.charCodeAt(n) & 255, r = /* Math.imul(h, m): */
      (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16);
  }
  return r ^= r >>> 13, r = /* Math.imul(h, m): */
  (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16), ((r ^ r >>> 15) >>> 0).toString(36);
}
var io = {
  animationIterationCount: 1,
  aspectRatio: 1,
  borderImageOutset: 1,
  borderImageSlice: 1,
  borderImageWidth: 1,
  boxFlex: 1,
  boxFlexGroup: 1,
  boxOrdinalGroup: 1,
  columnCount: 1,
  columns: 1,
  flex: 1,
  flexGrow: 1,
  flexPositive: 1,
  flexShrink: 1,
  flexNegative: 1,
  flexOrder: 1,
  gridRow: 1,
  gridRowEnd: 1,
  gridRowSpan: 1,
  gridRowStart: 1,
  gridColumn: 1,
  gridColumnEnd: 1,
  gridColumnSpan: 1,
  gridColumnStart: 1,
  msGridRow: 1,
  msGridRowSpan: 1,
  msGridColumn: 1,
  msGridColumnSpan: 1,
  fontWeight: 1,
  lineHeight: 1,
  opacity: 1,
  order: 1,
  orphans: 1,
  tabSize: 1,
  widows: 1,
  zIndex: 1,
  zoom: 1,
  WebkitLineClamp: 1,
  // SVG-related properties
  fillOpacity: 1,
  floodOpacity: 1,
  stopOpacity: 1,
  strokeDasharray: 1,
  strokeDashoffset: 1,
  strokeMiterlimit: 1,
  strokeOpacity: 1,
  strokeWidth: 1
};
function ao(e) {
  var r = /* @__PURE__ */ Object.create(null);
  return function(t) {
    return r[t] === void 0 && (r[t] = e(t)), r[t];
  };
}
var Jr = `You have illegal escape sequence in your template literal, most likely inside content's property value.
Because you write your CSS inside a JavaScript string you actually have to do double escaping, so for example "content: '\\00d7';" should become "content: '\\\\00d7';".
You can read more about this here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#ES2018_revision_of_illegal_escape_sequences`, so = "You have passed in falsy value as style object's key (can happen when in example you pass unexported component as computed key).", co = /[A-Z]|^ms/g, Ot = /_EMO_([^_]+?)_([^]*?)_EMO_/g, Tr = function(r) {
  return r.charCodeAt(1) === 45;
}, Xr = function(r) {
  return r != null && typeof r != "boolean";
}, cr = /* @__PURE__ */ ao(function(e) {
  return Tr(e) ? e : e.replace(co, "-$&").toLowerCase();
}), Ze = function(r, t) {
  switch (r) {
    case "animation":
    case "animationName":
      if (typeof t == "string")
        return t.replace(Ot, function(n, i, o) {
          return pe = {
            name: i,
            styles: o,
            next: pe
          }, i;
        });
  }
  return io[r] !== 1 && !Tr(r) && typeof t == "number" && t !== 0 ? t + "px" : t;
};
if (process.env.NODE_ENV !== "production") {
  var lo = /(var|attr|counters?|url|element|(((repeating-)?(linear|radial))|conic)-gradient)\(|(no-)?(open|close)-quote/, uo = ["normal", "none", "initial", "inherit", "unset"], fo = Ze, po = /^-ms-/, ho = /-(.)/g, Zr = {};
  Ze = function(r, t) {
    if (r === "content" && (typeof t != "string" || uo.indexOf(t) === -1 && !lo.test(t) && (t.charAt(0) !== t.charAt(t.length - 1) || t.charAt(0) !== '"' && t.charAt(0) !== "'")))
      throw new Error("You seem to be using a value for 'content' without quotes, try replacing it with `content: '\"" + t + "\"'`");
    var n = fo(r, t);
    return n !== "" && !Tr(r) && r.indexOf("-") !== -1 && Zr[r] === void 0 && (Zr[r] = !0, console.error("Using kebab-case for css properties in objects is not supported. Did you mean " + r.replace(po, "ms-").replace(ho, function(i, o) {
      return o.toUpperCase();
    }) + "?")), n;
  };
}
var _t = "Component selectors can only be used in conjunction with @emotion/babel-plugin, the swc Emotion plugin, or another Emotion-aware compiler transform.";
function Be(e, r, t) {
  if (t == null)
    return "";
  if (t.__emotion_styles !== void 0) {
    if (process.env.NODE_ENV !== "production" && t.toString() === "NO_COMPONENT_SELECTOR")
      throw new Error(_t);
    return t;
  }
  switch (typeof t) {
    case "boolean":
      return "";
    case "object": {
      if (t.anim === 1)
        return pe = {
          name: t.name,
          styles: t.styles,
          next: pe
        }, t.name;
      if (t.styles !== void 0) {
        var n = t.next;
        if (n !== void 0)
          for (; n !== void 0; )
            pe = {
              name: n.name,
              styles: n.styles,
              next: pe
            }, n = n.next;
        var i = t.styles + ";";
        return process.env.NODE_ENV !== "production" && t.map !== void 0 && (i += t.map), i;
      }
      return vo(e, r, t);
    }
    case "function": {
      if (e !== void 0) {
        var o = pe, s = t(e);
        return pe = o, Be(e, r, s);
      } else process.env.NODE_ENV !== "production" && console.error("Functions that are interpolated in css calls will be stringified.\nIf you want to have a css call based on props, create a function that returns a css call like this\nlet dynamicStyle = (props) => css`color: ${props.color}`\nIt can be called directly with props or interpolated in a styled call like this\nlet SomeComponent = styled('div')`${dynamicStyle}`");
      break;
    }
    case "string":
      if (process.env.NODE_ENV !== "production") {
        var c = [], d = t.replace(Ot, function(u, v, E) {
          var _ = "animation" + c.length;
          return c.push("const " + _ + " = keyframes`" + E.replace(/^@keyframes animation-\w+/, "") + "`"), "${" + _ + "}";
        });
        c.length && console.error("`keyframes` output got interpolated into plain string, please wrap it with `css`.\n\nInstead of doing this:\n\n" + [].concat(c, ["`" + d + "`"]).join(`
`) + `

You should wrap it with \`css\` like this:

` + ("css`" + d + "`"));
      }
      break;
  }
  if (r == null)
    return t;
  var y = r[t];
  return y !== void 0 ? y : t;
}
function vo(e, r, t) {
  var n = "";
  if (Array.isArray(t))
    for (var i = 0; i < t.length; i++)
      n += Be(e, r, t[i]) + ";";
  else
    for (var o in t) {
      var s = t[o];
      if (typeof s != "object")
        r != null && r[s] !== void 0 ? n += o + "{" + r[s] + "}" : Xr(s) && (n += cr(o) + ":" + Ze(o, s) + ";");
      else {
        if (o === "NO_COMPONENT_SELECTOR" && process.env.NODE_ENV !== "production")
          throw new Error(_t);
        if (Array.isArray(s) && typeof s[0] == "string" && (r == null || r[s[0]] === void 0))
          for (var c = 0; c < s.length; c++)
            Xr(s[c]) && (n += cr(o) + ":" + Ze(o, s[c]) + ";");
        else {
          var d = Be(e, r, s);
          switch (o) {
            case "animation":
            case "animationName": {
              n += cr(o) + ":" + d + ";";
              break;
            }
            default:
              process.env.NODE_ENV !== "production" && o === "undefined" && console.error(so), n += o + "{" + d + "}";
          }
        }
      }
    }
  return n;
}
var Qr = /label:\s*([^\s;\n{]+)\s*(;|$)/g, Ct;
process.env.NODE_ENV !== "production" && (Ct = /\/\*#\ssourceMappingURL=data:application\/json;\S+\s+\*\//g);
var pe, mo = function(r, t, n) {
  if (r.length === 1 && typeof r[0] == "object" && r[0] !== null && r[0].styles !== void 0)
    return r[0];
  var i = !0, o = "";
  pe = void 0;
  var s = r[0];
  s == null || s.raw === void 0 ? (i = !1, o += Be(n, t, s)) : (process.env.NODE_ENV !== "production" && s[0] === void 0 && console.error(Jr), o += s[0]);
  for (var c = 1; c < r.length; c++)
    o += Be(n, t, r[c]), i && (process.env.NODE_ENV !== "production" && s[c] === void 0 && console.error(Jr), o += s[c]);
  var d;
  process.env.NODE_ENV !== "production" && (o = o.replace(Ct, function(E) {
    return d = E, "";
  })), Qr.lastIndex = 0;
  for (var y = "", u; (u = Qr.exec(o)) !== null; )
    y += "-" + // $FlowFixMe we know it's not null
    u[1];
  var v = oo(o) + y;
  return process.env.NODE_ENV !== "production" ? {
    name: v,
    styles: o,
    map: d,
    next: pe,
    toString: function() {
      return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
    }
  } : {
    name: v,
    styles: o,
    next: pe
  };
}, yo = Zt, go = function(r) {
  return r !== "theme";
}, et = function(r) {
  return typeof r == "string" && // 96 is one less than the char code
  // for "a" so this is checking that
  // it's a lowercase character
  r.charCodeAt(0) > 96 ? yo : go;
}, rt = function(r, t, n) {
  var i;
  if (t) {
    var o = t.shouldForwardProp;
    i = r.__emotion_forwardProp && o ? function(s) {
      return r.__emotion_forwardProp(s) && o(s);
    } : o;
  }
  return typeof i != "function" && n && (i = r.__emotion_forwardProp), i;
}, tt = `You have illegal escape sequence in your template literal, most likely inside content's property value.
Because you write your CSS inside a JavaScript string you actually have to do double escaping, so for example "content: '\\00d7';" should become "content: '\\\\00d7';".
You can read more about this here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#ES2018_revision_of_illegal_escape_sequences`, bo = function(r) {
  var t = r.cache, n = r.serialized, i = r.isStringTag;
  return Rt(t, n, i), xt(function() {
    return no(t, n, i);
  }), null;
}, wo = function e(r, t) {
  if (process.env.NODE_ENV !== "production" && r === void 0)
    throw new Error(`You are trying to create a styled element with an undefined component.
You may have forgotten to import it.`);
  var n = r.__emotion_real === r, i = n && r.__emotion_base || r, o, s;
  t !== void 0 && (o = t.label, s = t.target);
  var c = rt(r, t, n), d = c || et(i), y = !d("as");
  return function() {
    var u = arguments, v = n && r.__emotion_styles !== void 0 ? r.__emotion_styles.slice(0) : [];
    if (o !== void 0 && v.push("label:" + o + ";"), u[0] == null || u[0].raw === void 0)
      v.push.apply(v, u);
    else {
      process.env.NODE_ENV !== "production" && u[0][0] === void 0 && console.error(tt), v.push(u[0][0]);
      for (var E = u.length, _ = 1; _ < E; _++)
        process.env.NODE_ENV !== "production" && u[0][_] === void 0 && console.error(tt), v.push(u[_], u[0][_]);
    }
    var S = St(function(m, x, T) {
      var R = y && m.as || i, N = "", W = [], p = m;
      if (m.theme == null) {
        p = {};
        for (var K in m)
          p[K] = m[K];
        p.theme = ie.useContext(Cr);
      }
      typeof m.className == "string" ? N = to(x.registered, W, m.className) : m.className != null && (N = m.className + " ");
      var O = mo(v.concat(W), x.registered, p);
      N += x.key + "-" + O.name, s !== void 0 && (N += " " + s);
      var Oe = y && c === void 0 ? et(R) : d, he = {};
      for (var ve in m)
        y && ve === "as" || // $FlowFixMe
        Oe(ve) && (he[ve] = m[ve]);
      return he.className = N, he.ref = T, /* @__PURE__ */ ie.createElement(ie.Fragment, null, /* @__PURE__ */ ie.createElement(bo, {
        cache: x,
        serialized: O,
        isStringTag: typeof R == "string"
      }), /* @__PURE__ */ ie.createElement(R, he));
    });
    return S.displayName = o !== void 0 ? o : "Styled(" + (typeof i == "string" ? i : i.displayName || i.name || "Component") + ")", S.defaultProps = r.defaultProps, S.__emotion_real = S, S.__emotion_base = i, S.__emotion_styles = v, S.__emotion_forwardProp = c, Object.defineProperty(S, "toString", {
      value: function() {
        return s === void 0 && process.env.NODE_ENV !== "production" ? "NO_COMPONENT_SELECTOR" : "." + s;
      }
    }), S.withComponent = function(m, x) {
      return e(m, vr({}, t, x, {
        shouldForwardProp: rt(S, x, !0)
      })).apply(void 0, v);
    }, S;
  };
}, Eo = [
  "a",
  "abbr",
  "address",
  "area",
  "article",
  "aside",
  "audio",
  "b",
  "base",
  "bdi",
  "bdo",
  "big",
  "blockquote",
  "body",
  "br",
  "button",
  "canvas",
  "caption",
  "cite",
  "code",
  "col",
  "colgroup",
  "data",
  "datalist",
  "dd",
  "del",
  "details",
  "dfn",
  "dialog",
  "div",
  "dl",
  "dt",
  "em",
  "embed",
  "fieldset",
  "figcaption",
  "figure",
  "footer",
  "form",
  "h1",
  "h2",
  "h3",
  "h4",
  "h5",
  "h6",
  "head",
  "header",
  "hgroup",
  "hr",
  "html",
  "i",
  "iframe",
  "img",
  "input",
  "ins",
  "kbd",
  "keygen",
  "label",
  "legend",
  "li",
  "link",
  "main",
  "map",
  "mark",
  "marquee",
  "menu",
  "menuitem",
  "meta",
  "meter",
  "nav",
  "noscript",
  "object",
  "ol",
  "optgroup",
  "option",
  "output",
  "p",
  "param",
  "picture",
  "pre",
  "progress",
  "q",
  "rp",
  "rt",
  "ruby",
  "s",
  "samp",
  "script",
  "section",
  "select",
  "small",
  "source",
  "span",
  "strong",
  "style",
  "sub",
  "summary",
  "sup",
  "table",
  "tbody",
  "td",
  "textarea",
  "tfoot",
  "th",
  "thead",
  "time",
  "title",
  "tr",
  "track",
  "u",
  "ul",
  "var",
  "video",
  "wbr",
  // SVG
  "circle",
  "clipPath",
  "defs",
  "ellipse",
  "foreignObject",
  "g",
  "image",
  "line",
  "linearGradient",
  "mask",
  "path",
  "pattern",
  "polygon",
  "polyline",
  "radialGradient",
  "rect",
  "stop",
  "svg",
  "text",
  "tspan"
], ne = wo.bind();
Eo.forEach(function(e) {
  ne[e] = ne(e);
});
var br = { exports: {} };
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/
var lr, nt;
function xo() {
  if (nt) return lr;
  nt = 1;
  var e = Object.getOwnPropertySymbols, r = Object.prototype.hasOwnProperty, t = Object.prototype.propertyIsEnumerable;
  function n(o) {
    if (o == null)
      throw new TypeError("Object.assign cannot be called with null or undefined");
    return Object(o);
  }
  function i() {
    try {
      if (!Object.assign)
        return !1;
      var o = new String("abc");
      if (o[5] = "de", Object.getOwnPropertyNames(o)[0] === "5")
        return !1;
      for (var s = {}, c = 0; c < 10; c++)
        s["_" + String.fromCharCode(c)] = c;
      var d = Object.getOwnPropertyNames(s).map(function(u) {
        return s[u];
      });
      if (d.join("") !== "0123456789")
        return !1;
      var y = {};
      return "abcdefghijklmnopqrst".split("").forEach(function(u) {
        y[u] = u;
      }), Object.keys(Object.assign({}, y)).join("") === "abcdefghijklmnopqrst";
    } catch {
      return !1;
    }
  }
  return lr = i() ? Object.assign : function(o, s) {
    for (var c, d = n(o), y, u = 1; u < arguments.length; u++) {
      c = Object(arguments[u]);
      for (var v in c)
        r.call(c, v) && (d[v] = c[v]);
      if (e) {
        y = e(c);
        for (var E = 0; E < y.length; E++)
          t.call(c, y[E]) && (d[y[E]] = c[y[E]]);
      }
    }
    return d;
  }, lr;
}
var ur, ot;
function Pr() {
  if (ot) return ur;
  ot = 1;
  var e = "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED";
  return ur = e, ur;
}
var dr, it;
function So() {
  if (it) return dr;
  it = 1;
  var e = function() {
  };
  if (process.env.NODE_ENV !== "production") {
    var r = Pr(), t = {}, n = Function.call.bind(Object.prototype.hasOwnProperty);
    e = function(o) {
      var s = "Warning: " + o;
      typeof console < "u" && console.error(s);
      try {
        throw new Error(s);
      } catch {
      }
    };
  }
  function i(o, s, c, d, y) {
    if (process.env.NODE_ENV !== "production") {
      for (var u in o)
        if (n(o, u)) {
          var v;
          try {
            if (typeof o[u] != "function") {
              var E = Error(
                (d || "React class") + ": " + c + " type `" + u + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof o[u] + "`."
              );
              throw E.name = "Invariant Violation", E;
            }
            v = o[u](s, u, d, c, null, r);
          } catch (S) {
            v = S;
          }
          if (v && !(v instanceof Error) && e(
            (d || "React class") + ": type specification of " + c + " `" + u + "` is invalid; the type checker function must return `null` or an `Error` but returned a " + typeof v + ". You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument)."
          ), v instanceof Error && !(v.message in t)) {
            t[v.message] = !0;
            var _ = y ? y() : "";
            e(
              "Failed " + c + " type: " + v.message + (_ ?? "")
            );
          }
        }
    }
  }
  return i.resetWarningCache = function() {
    process.env.NODE_ENV !== "production" && (t = {});
  }, dr = i, dr;
}
var fr, at;
function Ro() {
  if (at) return fr;
  at = 1;
  var e = yt, r = xo(), t = Pr(), n = So(), i = Function.call.bind(Object.prototype.hasOwnProperty), o = function() {
  };
  process.env.NODE_ENV !== "production" && (o = function(c) {
    var d = "Warning: " + c;
    typeof console < "u" && console.error(d);
    try {
      throw new Error(d);
    } catch {
    }
  });
  function s() {
    return null;
  }
  return fr = function(c, d) {
    var y = typeof Symbol == "function" && Symbol.iterator, u = "@@iterator";
    function v(f) {
      var w = f && (y && f[y] || f[u]);
      if (typeof w == "function")
        return w;
    }
    var E = "<<anonymous>>", _ = {
      array: T("array"),
      bool: T("boolean"),
      func: T("function"),
      number: T("number"),
      object: T("object"),
      string: T("string"),
      symbol: T("symbol"),
      any: R(),
      arrayOf: N,
      element: W(),
      elementType: p(),
      instanceOf: K,
      node: ve(),
      objectOf: Oe,
      oneOf: O,
      oneOfType: he,
      shape: _e,
      exact: Ie
    };
    function S(f, w) {
      return f === w ? f !== 0 || 1 / f === 1 / w : f !== f && w !== w;
    }
    function m(f) {
      this.message = f, this.stack = "";
    }
    m.prototype = Error.prototype;
    function x(f) {
      if (process.env.NODE_ENV !== "production")
        var w = {}, P = 0;
      function A(M, $, C, L, Y, U, ue) {
        if (L = L || E, U = U || C, ue !== t) {
          if (d) {
            var se = new Error(
              "Calling PropTypes validators directly is not supported by the `prop-types` package. Use `PropTypes.checkPropTypes()` to call them. Read more at http://fb.me/use-check-prop-types"
            );
            throw se.name = "Invariant Violation", se;
          } else if (process.env.NODE_ENV !== "production" && typeof console < "u") {
            var g = L + ":" + C;
            !w[g] && // Avoid spamming the console because they are often not actionable except for lib authors
            P < 3 && (o(
              "You are manually calling a React.PropTypes validation function for the `" + U + "` prop on `" + L + "`. This is deprecated and will throw in the standalone `prop-types` package. You may be seeing this warning due to a third-party PropTypes library. See https://fb.me/react-warning-dont-call-proptypes for details."
            ), w[g] = !0, P++);
          }
        }
        return $[C] == null ? M ? $[C] === null ? new m("The " + Y + " `" + U + "` is marked as required " + ("in `" + L + "`, but its value is `null`.")) : new m("The " + Y + " `" + U + "` is marked as required in " + ("`" + L + "`, but its value is `undefined`.")) : null : f($, C, L, Y, U);
      }
      var k = A.bind(null, !1);
      return k.isRequired = A.bind(null, !0), k;
    }
    function T(f) {
      function w(P, A, k, M, $, C) {
        var L = P[A], Y = le(L);
        if (Y !== f) {
          var U = Q(L);
          return new m("Invalid " + M + " `" + $ + "` of type " + ("`" + U + "` supplied to `" + k + "`, expected ") + ("`" + f + "`."));
        }
        return null;
      }
      return x(w);
    }
    function R() {
      return x(s);
    }
    function N(f) {
      function w(P, A, k, M, $) {
        if (typeof f != "function")
          return new m("Property `" + $ + "` of component `" + k + "` has invalid PropType notation inside arrayOf.");
        var C = P[A];
        if (!Array.isArray(C)) {
          var L = le(C);
          return new m("Invalid " + M + " `" + $ + "` of type " + ("`" + L + "` supplied to `" + k + "`, expected an array."));
        }
        for (var Y = 0; Y < C.length; Y++) {
          var U = f(C, Y, k, M, $ + "[" + Y + "]", t);
          if (U instanceof Error)
            return U;
        }
        return null;
      }
      return x(w);
    }
    function W() {
      function f(w, P, A, k, M) {
        var $ = w[P];
        if (!c($)) {
          var C = le($);
          return new m("Invalid " + k + " `" + M + "` of type " + ("`" + C + "` supplied to `" + A + "`, expected a single ReactElement."));
        }
        return null;
      }
      return x(f);
    }
    function p() {
      function f(w, P, A, k, M) {
        var $ = w[P];
        if (!e.isValidElementType($)) {
          var C = le($);
          return new m("Invalid " + k + " `" + M + "` of type " + ("`" + C + "` supplied to `" + A + "`, expected a single ReactElement type."));
        }
        return null;
      }
      return x(f);
    }
    function K(f) {
      function w(P, A, k, M, $) {
        if (!(P[A] instanceof f)) {
          var C = f.name || E, L = de(P[A]);
          return new m("Invalid " + M + " `" + $ + "` of type " + ("`" + L + "` supplied to `" + k + "`, expected ") + ("instance of `" + C + "`."));
        }
        return null;
      }
      return x(w);
    }
    function O(f) {
      if (!Array.isArray(f))
        return process.env.NODE_ENV !== "production" && (arguments.length > 1 ? o(
          "Invalid arguments supplied to oneOf, expected an array, got " + arguments.length + " arguments. A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z])."
        ) : o("Invalid argument supplied to oneOf, expected an array.")), s;
      function w(P, A, k, M, $) {
        for (var C = P[A], L = 0; L < f.length; L++)
          if (S(C, f[L]))
            return null;
        var Y = JSON.stringify(f, function(ue, se) {
          var g = Q(se);
          return g === "symbol" ? String(se) : se;
        });
        return new m("Invalid " + M + " `" + $ + "` of value `" + String(C) + "` " + ("supplied to `" + k + "`, expected one of " + Y + "."));
      }
      return x(w);
    }
    function Oe(f) {
      function w(P, A, k, M, $) {
        if (typeof f != "function")
          return new m("Property `" + $ + "` of component `" + k + "` has invalid PropType notation inside objectOf.");
        var C = P[A], L = le(C);
        if (L !== "object")
          return new m("Invalid " + M + " `" + $ + "` of type " + ("`" + L + "` supplied to `" + k + "`, expected an object."));
        for (var Y in C)
          if (i(C, Y)) {
            var U = f(C, Y, k, M, $ + "." + Y, t);
            if (U instanceof Error)
              return U;
          }
        return null;
      }
      return x(w);
    }
    function he(f) {
      if (!Array.isArray(f))
        return process.env.NODE_ENV !== "production" && o("Invalid argument supplied to oneOfType, expected an instance of array."), s;
      for (var w = 0; w < f.length; w++) {
        var P = f[w];
        if (typeof P != "function")
          return o(
            "Invalid argument supplied to oneOfType. Expected an array of check functions, but received " + Ce(P) + " at index " + w + "."
          ), s;
      }
      function A(k, M, $, C, L) {
        for (var Y = 0; Y < f.length; Y++) {
          var U = f[Y];
          if (U(k, M, $, C, L, t) == null)
            return null;
        }
        return new m("Invalid " + C + " `" + L + "` supplied to " + ("`" + $ + "`."));
      }
      return x(A);
    }
    function ve() {
      function f(w, P, A, k, M) {
        return xe(w[P]) ? null : new m("Invalid " + k + " `" + M + "` supplied to " + ("`" + A + "`, expected a ReactNode."));
      }
      return x(f);
    }
    function _e(f) {
      function w(P, A, k, M, $) {
        var C = P[A], L = le(C);
        if (L !== "object")
          return new m("Invalid " + M + " `" + $ + "` of type `" + L + "` " + ("supplied to `" + k + "`, expected `object`."));
        for (var Y in f) {
          var U = f[Y];
          if (U) {
            var ue = U(C, Y, k, M, $ + "." + Y, t);
            if (ue)
              return ue;
          }
        }
        return null;
      }
      return x(w);
    }
    function Ie(f) {
      function w(P, A, k, M, $) {
        var C = P[A], L = le(C);
        if (L !== "object")
          return new m("Invalid " + M + " `" + $ + "` of type `" + L + "` " + ("supplied to `" + k + "`, expected `object`."));
        var Y = r({}, P[A], f);
        for (var U in Y) {
          var ue = f[U];
          if (!ue)
            return new m(
              "Invalid " + M + " `" + $ + "` key `" + U + "` supplied to `" + k + "`.\nBad object: " + JSON.stringify(P[A], null, "  ") + `
Valid keys: ` + JSON.stringify(Object.keys(f), null, "  ")
            );
          var se = ue(C, U, k, M, $ + "." + U, t);
          if (se)
            return se;
        }
        return null;
      }
      return x(w);
    }
    function xe(f) {
      switch (typeof f) {
        case "number":
        case "string":
        case "undefined":
          return !0;
        case "boolean":
          return !f;
        case "object":
          if (Array.isArray(f))
            return f.every(xe);
          if (f === null || c(f))
            return !0;
          var w = v(f);
          if (w) {
            var P = w.call(f), A;
            if (w !== f.entries) {
              for (; !(A = P.next()).done; )
                if (!xe(A.value))
                  return !1;
            } else
              for (; !(A = P.next()).done; ) {
                var k = A.value;
                if (k && !xe(k[1]))
                  return !1;
              }
          } else
            return !1;
          return !0;
        default:
          return !1;
      }
    }
    function je(f, w) {
      return f === "symbol" ? !0 : w ? w["@@toStringTag"] === "Symbol" || typeof Symbol == "function" && w instanceof Symbol : !1;
    }
    function le(f) {
      var w = typeof f;
      return Array.isArray(f) ? "array" : f instanceof RegExp ? "object" : je(w, f) ? "symbol" : w;
    }
    function Q(f) {
      if (typeof f > "u" || f === null)
        return "" + f;
      var w = le(f);
      if (w === "object") {
        if (f instanceof Date)
          return "date";
        if (f instanceof RegExp)
          return "regexp";
      }
      return w;
    }
    function Ce(f) {
      var w = Q(f);
      switch (w) {
        case "array":
        case "object":
          return "an " + w;
        case "boolean":
        case "date":
        case "regexp":
          return "a " + w;
        default:
          return w;
      }
    }
    function de(f) {
      return !f.constructor || !f.constructor.name ? E : f.constructor.name;
    }
    return _.checkPropTypes = n, _.resetWarningCache = n.resetWarningCache, _.PropTypes = _, _;
  }, fr;
}
var pr, st;
function Oo() {
  if (st) return pr;
  st = 1;
  var e = Pr();
  function r() {
  }
  function t() {
  }
  return t.resetWarningCache = r, pr = function() {
    function n(s, c, d, y, u, v) {
      if (v !== e) {
        var E = new Error(
          "Calling PropTypes validators directly is not supported by the `prop-types` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types"
        );
        throw E.name = "Invariant Violation", E;
      }
    }
    n.isRequired = n;
    function i() {
      return n;
    }
    var o = {
      array: n,
      bool: n,
      func: n,
      number: n,
      object: n,
      string: n,
      symbol: n,
      any: n,
      arrayOf: i,
      element: n,
      elementType: n,
      instanceOf: i,
      node: n,
      objectOf: i,
      oneOf: i,
      oneOfType: i,
      shape: i,
      exact: i,
      checkPropTypes: t,
      resetWarningCache: r
    };
    return o.PropTypes = o, o;
  }, pr;
}
if (process.env.NODE_ENV !== "production") {
  var _o = yt, Co = !0;
  br.exports = Ro()(_o.isElement, Co);
} else
  br.exports = Oo()();
var we = br.exports;
const l = /* @__PURE__ */ Ht(we);
class Tt extends Ee.Component {
  constructor() {
    super(...arguments);
    F(this, "container", Ee.createRef());
    F(this, "handleClick", (t) => {
      const n = this.container.current, { target: i } = t, { onClickOutside: o } = this.props;
      (n && n === i || n && !n.contains(i)) && o(t);
    });
  }
  componentDidMount() {
    document.addEventListener("click", this.handleClick, !0);
  }
  componentWillUnmount() {
    document.removeEventListener("click", this.handleClick, !0);
  }
  render() {
    const { className: t, children: n } = this.props;
    return /* @__PURE__ */ j.jsx("div", { className: t, ref: this.container, children: n });
  }
}
Tt.propTypes = {
  onClickOutside: l.func.isRequired,
  children: l.node.isRequired,
  className: l.string
};
const Qe = (e, r, t) => !!r.find((n) => re(n, t.valueField) === e || re(n, t.labelField) === e), Ne = (e, r) => {
  e.length === 4 && (e = `${e[1]}${e[1]}${e[2]}${e[2]}${e[3]}${e[3]}}`);
  const t = parseInt(e.slice(1, 3), 16), n = parseInt(e.slice(3, 5), 16), i = parseInt(e.slice(5, 7), 16);
  return `rgba(${t}, ${n}, ${i}${r && `, ${r}`})`;
}, Ue = (e, r = 0) => {
  let t;
  return (...n) => {
    t && clearTimeout(t), t = setTimeout(() => {
      e(...n), t = null;
    }, r);
  };
}, To = (e, r) => JSON.stringify(e) === JSON.stringify(r), re = (e, r) => {
  if (r)
    return r.split(".").reduce((t, n) => t[n], e);
}, Fe = (e, r, t) => {
  if (!r)
    return e;
  const n = Array.isArray(r) ? r : r.split(".").filter((i) => i.length);
  return n.length ? Fe(e[n.shift()], n, t) : e === void 0 ? t : e;
}, Ae = () => (typeof window > "u" && (global.window = {}), window), G = "react-dropdown-select", Po = ({ item: e, props: r, state: t, methods: n }) => e && r.optionRenderer ? r.optionRenderer({ item: e, props: r, state: t, methods: n }) : /* @__PURE__ */ j.jsxs(
  Ao,
  {
    role: "listitem",
    disabled: r.disabled,
    direction: r.direction,
    className: `${G}-option`,
    color: r.color,
    children: [
      /* @__PURE__ */ j.jsx("span", { className: `${G}-option-label`, children: re(e, r.labelField) }),
      /* @__PURE__ */ j.jsx(
        "span",
        {
          className: `${G}-option-remove`,
          onClick: (i) => n.removeItem(i, e, r.closeOnSelect),
          children: ""
        }
      )
    ]
  }
), Ao = ne.span`
  padding: 0 5px;
  border-radius: 2px;
  line-height: 21px;
  margin: 3px 0 3px 5px;
  background: ${({ color: e }) => e};
  color: #fff;
  display: flex;
  flex-direction: ${({ direction: e }) => e === "rtl" ? "row-reverse" : "row"};
  

  .${G}-option-remove {
    cursor: pointer;
    width: 22px;
    height: 22px;
    display: inline-block;
    text-align: center;
    margin: 0 -5px 0 0px;
    border-radius: 0 3px 3px 0;

    :hover {
      color: tomato;
    }
  }

  :hover,
  :hover > span {
    opacity: 0.9;
  }
`, ko = (e, r) => {
  const { addPlaceholder: t, searchable: n, placeholder: i } = e, o = r.values && r.values.length === 0, s = r.values && r.values.length > 0;
  return s && t && n ? t : o ? i : "";
};
class Pt extends wr {
  constructor() {
    super(...arguments);
    F(this, "input", Ee.createRef());
    F(this, "onBlur", (t) => (t.stopPropagation(), this.props.state.dropdown ? this.input.current.focus() : this.input.current.blur()));
    F(this, "handleKeyPress", (t) => {
      const { props: n, state: i, methods: o } = this.props;
      return n.create && t.key === "Enter" && !Qe(i.search, [...i.values, ...n.options], this.props) && i.search && i.cursor === null && o.createNew(i.search);
    });
  }
  componentDidUpdate(t) {
    (this.props.state.dropdown || t.state.dropdown !== this.props.state.dropdown && this.props.state.dropdown || this.props.props.autoFocus) && this.input.current.focus(), t.state.dropdown !== this.props.state.dropdown && !this.props.state.dropdown && this.input.current.blur();
  }
  render() {
    const { props: t, state: n, methods: i } = this.props;
    return t.inputRenderer ? t.inputRenderer({ props: t, state: n, methods: i, inputRef: this.input }) : /* @__PURE__ */ j.jsx(
      $o,
      {
        ref: this.input,
        tabIndex: "-1",
        onFocus: (o) => o.stopPropagation(),
        className: `${G}-input`,
        size: i.getInputSize(),
        value: n.search,
        readOnly: !t.searchable,
        onClick: () => i.dropDown("open"),
        onKeyPress: this.handleKeyPress,
        onChange: i.setSearch,
        onBlur: this.onBlur,
        placeholder: ko(t, n),
        disabled: t.disabled
      }
    );
  }
}
Pt.propTypes = {
  props: we.object,
  state: we.object,
  methods: we.object
};
const $o = ne.input`
  line-height: inherit;
  border: none;
  margin-left: 5px;
  background: transparent;
  padding: 0;
  width: calc(${({ size: e }) => `${e}ch`} + 5px);
  font-size: smaller;
  ${({ readOnly: e }) => e && "cursor: pointer;"}
  :focus {
    outline: none;
  }
`, At = Object.freeze({
  /**
   * Secondary placeholder on search field if any value selected
   */
  addPlaceholder: l.string,
  /**
   * Additional props to pass to Select
   */
  additionalProps: l.object,
  /**
   * If true, and searchable, dropdown will autofocus
   */
  autoFocus: l.bool,
  /**
   * If true, backspace key will delete last value
   */
  backspaceDelete: l.bool,
  /**
   * CSS class attribute to pass to select
   */
  className: l.string,
  /**
   * Label for "Clear all"
   */
  clearAllLabel: l.string,
  /**
   * If true, and searchable, search value will be cleared on blur
   */
  clearOnBlur: l.bool,
  /**
   * If true, and searchable, search value will be cleared upon value select/de-select
   */
  clearOnSelect: l.bool,
  /**
   * Overrides internal clear button
   */
  clearRenderer: l.func,
  /**
   * Clear all indicator
   */
  clearable: l.bool,
  /**
   * If true, scrolling the page will close the dropdown
   */
  closeOnScroll: l.bool,
  /**
   * If true, selecting option will close the dropdown
   */
  closeOnSelect: l.bool,
  /**
   * If true, clicking input will close the dropdown if you are not searching.
   */
  closeOnClickInput: l.bool,
  /**
   * Base color (any CSS compatible) to use in component, also can be overwritten via CSS
   */
  color: l.string,
  /**
   * Compare values override function
   */
  compareValuesFunc: l.func,
  /**
   * Overrides internal content component (the contents of the select component)
   * | <a href="https://sanusart.github.io/react-dropdown-select/prop/content-renderer">example</a>
   */
  contentRenderer: l.func,
  /**
   * If true, select will create value from search string and fire onCreateNew callback prop
   */
  create: l.bool,
  /**
   * If create set to true, this will be the label of the "add new" component. {search} will be replaced by search value
   */
  createNewLabel: l.string,
  /**
   * Debounce Delay for updates upon component interactions
   */
  debounceDelay: l.number,
  /**
   * Direction of a dropdown "ltr", "rtl" or "auto"
   */
  direction: l.string,
  /**
   * Disable select and all interactions
   */
  disabled: l.bool,
  /**
   * Label shown on disabled field (after) the text
   */
  disabledLabel: l.string,
  /**
   * Gap between select element and dropdown
   */
  dropdownGap: l.number,
  /**
   * Show or hide dropdown handle to open/close dropdown
   */
  dropdownHandle: l.bool,
  /**
   * Overrides internal dropdown handle
   */
  dropdownHandleRenderer: l.func,
  /**
   * Minimum height of a dropdown
   */
  dropdownHeight: l.string,
  /**
   * Available options are "auto", "top" and "bottom" defaults to "bottom". Auto will adjust itself according Select's position on the page
   * | <a href="https://sanusart.github.io/react-dropdown-select/prop/dropdown-position">example</a>
   */
  dropdownPosition: l.oneOf(["top", "bottom", "auto"]),
  /**
   * Overrides internal dropdown handle
   */
  dropdownRenderer: l.func,
  /**
   * Overrides internal keyDown function
   */
  handleKeyDownFn: l.func,
  /**
   * Overrides internal input text
   */
  inputRenderer: l.func,
  /**
   * Overrides internal item in a dropdown
   */
  itemRenderer: l.func,
  /**
   * If true, dropdown will always stay open (good for debugging)
   */
  keepOpen: l.bool,
  /**
   * If false, selected item will not appear in a list
   */
  keepSelectedInList: l.bool,
  /**
   * Field in data to use for label
   */
  labelField: l.string,
  /**
   * Loading indicator
   */
  loading: l.bool,
  /**
   * Overrides internal loading
   */
  loadingRenderer: l.func,
  /**
   * If true - will act as multi-select, if false - only one option will be selected at the time
   */
  multi: l.bool,
  /**
   * If set, input type hidden would be added in the component with the value of the `name` prop as `name` and select's `values` as `value`
   * Useful for HTML forms
   */
  name: l.string,
  /**
   * Label for "No data"
   */
  noDataLabel: l.string,
  /**
   * Overrides internal "no data" (shown where search has no results)
   */
  noDataRenderer: l.func,
  /**
   * onChange callback handler
   */
  onChange: l.func.isRequired,
  /**
   * Fires upon clearing all values (via custom renderers)
   */
  onClearAll: l.func,
  /**
   * Fires upon creation of new item if create prop set to true
   */
  onCreateNew: l.func,
  /**
   * Fires upon dropdown close
   */
  onDropdownClose: l.func,
  /**
   * Fires upon dropdown closing state, stops the closing and provides own method close()
   * @return undefined
   */
  onDropdownCloseRequest: l.func,
  /**
   * Fires upon dropdown open
   */
  onDropdownOpen: l.func,
  /**
   * Fires upon selecting all values (via custom renderers)
   */
  onSelectAll: l.func,
  /**
   * Overrides internal option (the pillow with an "x") on the select content
   */
  optionRenderer: l.func,
  /**
   * Available options, (option with key disabled: true will be disabled)
   */
  options: l.array.isRequired,
  /**
   * If set, input type hidden would be added in the component with pattern prop as regex
   */
  pattern: l.string,
  /**
   * Placeholder shown where there are no selected values
   */
  placeholder: l.string,
  /**
   * If valid DOM element specified - dropdown will break out to render inside the specified element
   */
  portal: l.element,
  /**
   * If set, input type hidden would be added in the component with required prop as true/false
   */
  required: l.bool,
  /**
   * Search by object property in values
   */
  searchBy: l.string,
  /**
   * Overrides internal search function
   */
  searchFn: l.func,
  /**
   * If true, select will have search input text
   */
  searchable: l.bool,
  /**
   * Allow to select all (if select is multi-select)
   */
  selectAll: l.bool,
  /**
   * Label for "Select all"
   */
  selectAllLabel: l.string,
  /**
   * Separator line between close all and dropdown handle
   */
  separator: l.bool,
  /**
   * Overrides internal separator
   */
  separatorRenderer: l.func,
  /**
   * Sort by object property in values
   */
  sortBy: l.string,
  /**
   * Style object to pass to select
   */
  style: l.object,
  /**
   * Field in data to use for value
   */
  valueField: l.string,
  /**
   * Selected values
   */
  values: l.array,
  /**
   * If true, dropdown will be open by default
   */
  defaultMenuIsOpen: l.bool
}), No = Object.freeze({
  activeCursorItem: l.func,
  /**
   * Add a new item
   */
  addItem: l.func.isRequired,
  /**
   * Check if all items are selected
   */
  areAllSelected: l.func.isRequired,
  /**
   * Clear all selected items
   */
  clearAll: l.func.isRequired,
  /**
   * Create a new item
   */
  createNew: l.func.isRequired,
  /**
   * Close/Toggle/Open
   */
  dropDown: l.func.isRequired,
  /**
   * Get the input size
   */
  getInputSize: l.func.isRequired,
  /**
   * Get the bounds of the select component
   */
  getSelectBounds: l.func.isRequired,
  /**
   * Get the reference to the select component
   */
  getSelectRef: l.func.isRequired,
  /**
   * Handle key down event
   */
  handleKeyDown: l.func.isRequired,
  /**
   * Check if an item is selected
   */
  isSelected: l.func.isRequired,
  /**
   * Remove an item
   */
  removeItem: l.func.isRequired,
  /**
   * Make a string safe
   */
  safeString: l.func.isRequired,
  /**
   * Get search results
   */
  searchResults: l.func.isRequired,
  /**
   * Select all items
   */
  selectAll: l.func.isRequired,
  /**
   * Set the search string
   */
  setSearch: l.func.isRequired,
  /**
   * Sort items
   */
  sortBy: l.func.isRequired,
  /**
   * Toggle select all
   */
  toggleSelectAll: l.func.isRequired
}), Do = Object.freeze({
  /**
   * Flag indicating whether the dropdown is open or closed
   */
  dropdown: l.bool.isRequired,
  /**
   * Array of selected values
   */
  values: l.arrayOf(l.shape({})),
  /**
   * Search string
   */
  search: l.string.isRequired,
  /**
   * Array of bounds for the select component
   */
  selectBounds: l.shape({}),
  /**
   * Cursor position
   */
  cursor: l.number,
  /**
   * Array of search results
   */
  searchResults: l.arrayOf(l.shape({}))
}), kt = ({ props: e, state: r, methods: t }) => /* @__PURE__ */ j.jsx(
  Io,
  {
    className: `${G}-content ${e.multi ? `${G}-type-multi` : `${G}-type-single`}`,
    onClick: (n) => (n.stopPropagation(), r.dropdown === !0 && e.closeOnClickInput && !r.search ? t.dropDown("close") : t.dropDown("open")),
    children: e.contentRenderer ? e.contentRenderer({ props: e, state: r, methods: t }) : /* @__PURE__ */ j.jsxs(Ee.Fragment, { children: [
      e.multi ? r.values && r.values.map((n) => /* @__PURE__ */ j.jsx(
        Po,
        {
          item: n,
          state: r,
          props: e,
          methods: t
        },
        `${re(n, e.valueField)}${re(n, e.labelField)}`
      )) : r.values && r.values.length > 0 && /* @__PURE__ */ j.jsx("span", { children: re(r.values[0], e.labelField) }),
      /* @__PURE__ */ j.jsx(Pt, { props: e, methods: t, state: r })
    ] })
  }
);
kt.propTypes = {
  props: l.shape(At),
  state: l.shape(Do),
  methods: l.shape(No)
};
const Io = ne.div`
  display: flex;
  flex: 1;
  flex-wrap: wrap;
`, jo = ({ props: e, state: r, methods: t }) => e.noDataRenderer ? e.noDataRenderer({ props: e, state: r, methods: t }) : /* @__PURE__ */ j.jsx(Lo, { className: `${G}-no-data`, color: e.color, children: e.noDataLabel }), Lo = ne.div`
  padding: 10px;
  text-align: center;
  color: ${({ color: e }) => e};
`;
class $t extends wr {
  constructor() {
    super(...arguments);
    F(this, "item", Ee.createRef());
  }
  componentDidMount() {
    const { props: t, methods: n } = this.props;
    this.item.current && !t.multi && t.keepSelectedInList && n.isSelected(this.props.item) && this.item.current.scrollIntoView({ block: "nearest", inline: "start" });
  }
  componentDidUpdate() {
    this.props.state.cursor === this.props.itemIndex && this.item.current && this.item.current.scrollIntoView({ behavior: "smooth", block: "nearest", inline: "start" });
  }
  render() {
    const { props: t, state: n, methods: i, item: o, itemIndex: s } = this.props;
    return t.itemRenderer ? t.itemRenderer({ item: o, itemIndex: s, props: t, state: n, methods: i }) : !t.keepSelectedInList && i.isSelected(o) ? null : /* @__PURE__ */ j.jsxs(
      Mo,
      {
        role: "option",
        ref: this.item,
        "aria-selected": i.isSelected(o),
        "aria-disabled": o.disabled,
        "aria-label": re(o, t.labelField),
        disabled: o.disabled,
        tabIndex: "-1",
        className: `${G}-item ${i.isSelected(o) ? `${G}-item-selected` : ""} ${n.cursor === s ? `${G}-item-active` : ""} ${o.disabled ? `${G}-item-disabled` : ""}`,
        onClick: o.disabled ? void 0 : () => i.addItem(o),
        onKeyPress: o.disabled ? void 0 : () => i.addItem(o),
        color: t.color,
        children: [
          re(o, t.labelField),
          " ",
          o.disabled && /* @__PURE__ */ j.jsx("ins", { children: t.disabledLabel })
        ]
      },
      `${re(o, t.valueField)}${re(o, t.labelField)}`
    );
  }
}
$t.propTypes = {
  props: we.any,
  state: we.any,
  methods: we.any,
  item: we.any,
  itemIndex: we.any
};
const Mo = ne.span`
  padding: 5px 10px;
  cursor: pointer;
  border-bottom: 1px solid #fff;

  &.${G}-item-active {
    border-bottom: 1px solid #fff;
    ${({ disabled: e, color: r }) => !e && r && `background: ${Ne(r, 0.1)};`}
  }

  :hover,
  :focus {
    background: ${({ color: e }) => e && Ne(e, 0.1)};
    outline: none;
  }

  &.${G}-item-selected {
    ${({ disabled: e, color: r }) => e ? `
    background: #f2f2f2;
    color: #ccc;
    ` : `
    background: ${r};
    color: #fff;
    border-bottom: 1px solid #fff;
    `}
  }

  ${({ disabled: e }) => e ? `
    background: #f2f2f2;
    color: #ccc;

    ins {
      text-decoration: none;
      border:1px solid #ccc;
      border-radius: 2px;
      padding: 0px 3px;
      font-size: x-small;
      text-transform: uppercase;
    }
    ` : ""}
`, ct = (e, r) => {
  const t = r.getSelectRef().getBoundingClientRect(), n = t.bottom + parseInt(e.dropdownHeight, 10) + parseInt(e.dropdownGap, 10);
  return e.dropdownPosition !== "auto" ? e.dropdownPosition : n > Ae().innerHeight && n > t.top ? "top" : "bottom";
}, lt = ({ props: e, state: r, methods: t }) => /* @__PURE__ */ j.jsx(
  Fo,
  {
    tabIndex: "-1",
    "aria-expanded": "true",
    role: "list",
    dropdownPosition: ct(e, t),
    selectBounds: r.selectBounds,
    portal: e.portal,
    dropdownGap: e.dropdownGap,
    dropdownHeight: e.dropdownHeight,
    className: `${G}-dropdown ${G}-dropdown-position-${ct(
      e,
      t
    )}`,
    children: e.dropdownRenderer ? e.dropdownRenderer({ props: e, state: r, methods: t }) : /* @__PURE__ */ j.jsxs(Ee.Fragment, { children: [
      e.create && r.search && !Qe(r.search, [...r.values, ...e.options], e) && /* @__PURE__ */ j.jsx(
        qo,
        {
          role: "button",
          className: `${G}-dropdown-add-new`,
          color: e.color,
          onClick: () => t.createNew(r.search),
          children: e.createNewLabel.replace("{search}", `"${r.search}"`)
        }
      ),
      r.searchResults.length === 0 ? /* @__PURE__ */ j.jsx(jo, { className: `${G}-no-data`, state: r, props: e, methods: t }) : r.searchResults.map((n, i) => /* @__PURE__ */ j.jsx(
        $t,
        {
          item: n,
          itemIndex: i,
          state: r,
          props: e,
          methods: t
        },
        n[e.valueField].toString()
      )),
      e.selectAll && e.options && e.multi && /* @__PURE__ */ j.jsx(
        Vo,
        {
          role: "button",
          className: `${G}-dropdown-select-all`,
          color: e.color,
          onClick: () => t.areAllSelected() ? t.clearAll() : t.selectAll(),
          children: t.areAllSelected() ? e.clearAllLabel : e.selectAllLabel
        }
      )
    ] })
  }
), Fo = ne.div`
  position: absolute;
  ${({ selectBounds: e, dropdownGap: r, dropdownPosition: t }) => t === "top" ? `bottom: ${e.height + 2 + r}px` : `top: ${e.height + 2 + r}px`};

  ${({ selectBounds: e, dropdownGap: r, dropdownPosition: t, portal: n }) => n ? `
      position: fixed;
      ${t === "bottom" ? `top: ${e.bottom + r}px;` : `bottom: ${Ae().innerHeight - e.top + r}px;`}
      left: ${e.left - 1}px;` : "left: -1px;"};
  border: 1px solid #ccc;
  width: ${({ selectBounds: e }) => e.width}px;
  padding: 0;
  display: flex;
  flex-direction: column;
  background: #fff;
  border-radius: 2px;
  box-shadow: 0 0 10px 0 ${() => Ne("#000000", 0.2)};
  max-height: ${({ dropdownHeight: e }) => e};
  overflow: auto;
  z-index: 9;

  :focus {
    outline: none;
  }
}
`, qo = ne.div`
  color: ${({ color: e }) => e};
  padding: 5px 10px;

  :hover {
    background: ${({ color: e }) => e && Ne(e, 0.1)};
    outline: none;
    cursor: pointer;
  }
`, Vo = ne.div`
  color: ${({ color: e }) => e};
  padding: 5px 10px;
  position: sticky;
  bottom: 0;
  margin: 0;
  opacity: 1;
  background: #fff;
  box-shadow: 0 0 10px 0 ${() => Ne("#000000", 0.2)};

  :hover {
    outline: none;
    cursor: pointer;
  }
`, Yo = ({ props: e }) => e.loadingRenderer ? e.loadingRenderer({ props: e }) : /* @__PURE__ */ j.jsx(Bo, { className: `${G}-loading`, color: e.color }), Bo = ne.div`
  @keyframes dual-ring-spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(180deg);
    }
  }

  padding: 0 5px;
  display: block;
  width: auto;
  height: auto;

  :after {
    content: ' ';
    display: block;
    width: 16px;
    height: 16px;
    border-radius: 50%;
    border-width: 1px;
    border-style: solid;
    border-color: ${({ color: e }) => e} transparent;
    animation: dual-ring-spin 0.7s ease-in-out infinite;
    margin: 0 0 0 -10px;
  }
`, zo = ({ props: e, state: r, methods: t }) => e.clearRenderer ? e.clearRenderer({ props: e, state: r, methods: t }) : /* @__PURE__ */ j.jsx(
  Wo,
  {
    className: `${G}-clear`,
    tabIndex: "-1",
    onClick: () => t.clearAll(),
    onKeyPress: () => t.clearAll(),
    children: ""
  }
), Wo = ne.div`
  line-height: 25px;
  margin: 0 10px;
  cursor: pointer;

  :focus {
    outline: none;
  }

  :hover {
    color: tomato;
  }
`, Uo = ({ props: e, state: r, methods: t }) => e.separatorRenderer ? e.separatorRenderer({ props: e, state: r, methods: t }) : /* @__PURE__ */ j.jsx(Ho, { className: `${G}-separator` }), Ho = ne.div`
  border-left: 1px solid #ccc;
  width: 1px;
  height: 25px;
  display: block;
`, Ko = ({ props: e, state: r, methods: t }) => /* @__PURE__ */ j.jsx(
  Go,
  {
    tabIndex: "-1",
    onClick: (n) => t.dropDown(r.dropdown ? "close" : "open", n),
    dropdownOpen: r.dropdown,
    onKeyPress: (n) => t.dropDown("toggle", n),
    onKeyDown: (n) => t.dropDown("toggle", n),
    className: `${G}-dropdown-handle`,
    rotate: e.dropdownHandleRenderer ? 0 : 1,
    color: e.color,
    children: e.dropdownHandleRenderer ? e.dropdownHandleRenderer({ props: e, state: r, methods: t }) : /* @__PURE__ */ j.jsx("svg", { fill: "currentColor", viewBox: "0 0 40 40", children: /* @__PURE__ */ j.jsx("path", { d: "M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z" }) })
  }
), Go = ne.div`
  text-align: center;
  ${({ dropdownOpen: e, rotate: r }) => e ? `
      pointer-events: all;
      ${r ? "transform: rotate(0deg);margin: 0px 0 -3px 5px;" : ""}
      ` : `
      pointer-events: none;
      ${r ? "margin: 0 0 0 5px;transform: rotate(180deg);" : ""}
      `};
  cursor: pointer;

  svg {
    width: 16px;
    height: 16px;
  }

  :hover {
    path {
      stroke: ${({ color: e }) => e};
    }
  }

  :focus {
    outline: none;

    path {
      stroke: ${({ color: e }) => e};
    }
  }
`;
class Nt extends wr {
  constructor(t) {
    super(t);
    F(this, "onDropdownClose", () => {
      this.setState({ cursor: null }), this.props.onDropdownClose();
    });
    F(this, "onScroll", () => {
      this.props.closeOnScroll && this.dropDown("close"), this.updateSelectBounds();
    });
    F(this, "updateSelectBounds", () => this.select.current && this.setState({
      selectBounds: this.select.current.getBoundingClientRect()
    }));
    F(this, "getSelectBounds", () => this.state.selectBounds);
    F(this, "dropDown", (t = "toggle", n, i = !1) => {
      const o = n && n.target || n && n.srcElement;
      if (this.props.onDropdownCloseRequest !== void 0 && this.state.dropdown && i === !1 && t === "close")
        return this.props.onDropdownCloseRequest({
          props: this.props,
          methods: this.methods,
          state: this.state,
          close: () => this.dropDown("close", null, !0)
        });
      if (!(this.props.portal && !this.props.closeOnScroll && !this.props.closeOnSelect && n && o && o.offsetParent && o.offsetParent.classList.contains("react-dropdown-select-dropdown")))
        return this.props.keepOpen ? this.setState({ dropdown: !0 }) : t === "close" && this.state.dropdown ? (this.select.current.blur(), this.setState({
          dropdown: !1,
          search: this.props.clearOnBlur ? "" : this.state.search,
          searchResults: this.props.options
        })) : t === "open" && !this.state.dropdown ? this.setState({ dropdown: !0 }) : t === "toggle" ? (this.select.current.focus(), this.setState({ dropdown: !this.state.dropdown })) : !1;
    });
    F(this, "getSelectRef", () => this.select.current);
    F(this, "addItem", (t) => {
      if (this.props.multi) {
        if (Qe(re(t, this.props.valueField), this.state.values, this.props))
          return this.removeItem(null, t, !1);
        this.setState({
          values: [...this.state.values, t]
        });
      } else
        this.setState({
          values: [t],
          dropdown: !1
        });
      return this.props.clearOnSelect && this.setState({ search: "" }, () => {
        this.setState({ searchResults: this.searchResults() });
      }), !0;
    });
    F(this, "removeItem", (t, n, i = !1) => {
      t && i && (t.preventDefault(), t.stopPropagation(), this.dropDown("close")), this.setState({
        values: this.state.values.filter(
          (o) => re(o, this.props.valueField) !== re(n, this.props.valueField)
        )
      });
    });
    F(this, "setSearch", (t) => {
      this.setState({
        cursor: null
      }), this.setState(
        {
          search: t.target.value
        },
        () => {
          this.setState({ searchResults: this.searchResults() });
        }
      );
    });
    F(this, "getInputSize", () => this.state.search ? this.state.search.length : this.state.values.length > 0 ? this.props.addPlaceholder.length : this.props.placeholder.length);
    F(this, "toggleSelectAll", () => this.setState({
      values: this.state.values.length === 0 ? this.selectAll() : this.clearAll()
    }));
    F(this, "clearAll", () => {
      this.props.onClearAll(), this.setState({
        values: []
      });
    });
    F(this, "selectAll", (t = []) => {
      this.props.onSelectAll();
      const n = t.length > 0 ? t : this.props.options.filter((i) => !i.disabled);
      this.setState({ values: n });
    });
    F(this, "isSelected", (t) => !!this.state.values.find(
      (n) => re(n, this.props.valueField) === re(t, this.props.valueField)
    ));
    F(this, "areAllSelected", () => this.state.values.length === this.props.options.filter((t) => !t.disabled).length);
    F(this, "safeString", (t) => t.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"));
    F(this, "sortBy", () => {
      const { sortBy: t, options: n } = this.props;
      return t && n.sort((i, o) => Fe(i, t) < Fe(o, t) ? -1 : Fe(i, t) > Fe(o, t) ? 1 : 0), n;
    });
    F(this, "searchFn", ({ state: t, methods: n }) => {
      const i = new RegExp(n.safeString(t.search), "i");
      return n.sortBy().filter(
        (o) => i.test(re(o, this.props.searchBy) || re(o, this.props.valueField))
      );
    });
    F(this, "searchResults", () => {
      const t = { state: this.state, props: this.props, methods: this.methods };
      return this.props.searchFn(t) || this.searchFn(t);
    });
    F(this, "activeCursorItem", (t) => this.setState({
      activeCursorItem: t
    }));
    F(this, "handleKeyDown", (t) => {
      const n = {
        event: t,
        state: this.state,
        props: this.props,
        methods: this.methods,
        setState: this.setState.bind(this)
      };
      return this.props.handleKeyDownFn(n) || this.handleKeyDownFn(n);
    });
    F(this, "handleKeyDownFn", ({ event: t, state: n, props: i, methods: o, setState: s }) => {
      const { cursor: c, searchResults: d } = n, y = t.key === "Escape", u = t.key === "Enter", v = t.key === "ArrowUp", E = t.key === "ArrowDown", _ = t.key === "Backspace", S = t.key === "Tab" && !t.shiftKey, m = t.shiftKey && t.key === "Tab";
      if (E && !n.dropdown)
        return t.preventDefault(), this.dropDown("open"), s({
          cursor: 0
        });
      if ((E || S && n.dropdown) && c === null)
        return s({
          cursor: 0
        });
      if ((v || E || m && n.dropdown || S && n.dropdown) && t.preventDefault(), y && this.dropDown("close"), u) {
        const x = d[c];
        if (x && !x.disabled) {
          if (i.create && Qe(n.search, n.values, i))
            return null;
          o.addItem(x);
        }
      }
      if ((E || S && n.dropdown) && d.length === c)
        return s({
          cursor: 0
        });
      (E || S && n.dropdown) && s((x) => ({
        cursor: x.cursor + 1
      })), (v || m && n.dropdown) && c > 0 && s((x) => ({
        cursor: x.cursor - 1
      })), (v || m && n.dropdown) && c === 0 && s({
        cursor: d.length
      }), _ && i.backspaceDelete && this.getInputSize() === 0 && this.setState({
        values: this.state.values.slice(0, -1)
      });
    });
    F(this, "renderDropdown", () => this.props.portal ? Ut.createPortal(
      /* @__PURE__ */ j.jsx(lt, { props: this.props, state: this.state, methods: this.methods }),
      this.dropdownRoot
    ) : /* @__PURE__ */ j.jsx(lt, { props: this.props, state: this.state, methods: this.methods }));
    F(this, "createNew", (t) => {
      const n = {
        [this.props.labelField]: t,
        [this.props.valueField]: t
      };
      this.addItem(n), this.props.onCreateNew(n), this.setState({ search: "" });
    });
    this.state = {
      dropdown: !1,
      values: t.values,
      search: "",
      selectBounds: {},
      cursor: null,
      searchResults: t.options
    }, this.methods = {
      activeCursorItem: this.activeCursorItem,
      addItem: this.addItem,
      areAllSelected: this.areAllSelected,
      clearAll: this.clearAll,
      createNew: this.createNew,
      dropDown: this.dropDown,
      getInputSize: this.getInputSize,
      getSelectBounds: this.getSelectBounds,
      getSelectRef: this.getSelectRef,
      handleKeyDown: this.handleKeyDown,
      isSelected: this.isSelected,
      removeItem: this.removeItem,
      safeString: this.safeString,
      searchResults: this.searchResults,
      selectAll: this.selectAll,
      setSearch: this.setSearch,
      sortBy: this.sortBy,
      toggleSelectAll: this.toggleSelectAll
    }, this.select = Ee.createRef(), this.dropdownRoot = typeof document < "u" && document.createElement("div");
  }
  componentDidMount() {
    this.props.portal && this.props.portal.appendChild(this.dropdownRoot), Ae().addEventListener("resize", Ue(this.updateSelectBounds)), Ae().addEventListener("scroll", Ue(this.onScroll)), this.dropDown("close"), this.select && this.updateSelectBounds(), this.props.defaultMenuIsOpen && this.dropDown("open");
  }
  componentDidUpdate(t, n) {
    !this.props.compareValuesFunc(t.values, this.props.values) && this.props.compareValuesFunc(t.values, n.values) && (this.setState(
      {
        values: this.props.values
      },
      () => {
        this.props.onChange(this.state.values);
      }
    ), this.updateSelectBounds()), t.options !== this.props.options && this.setState({ searchResults: this.searchResults() }), n.values !== this.state.values && (this.props.onChange(this.state.values), this.updateSelectBounds()), n.search !== this.state.search && this.updateSelectBounds(), n.values !== this.state.values && this.props.closeOnSelect && this.dropDown("close"), t.multi !== this.props.multi && this.updateSelectBounds(), n.dropdown && n.dropdown !== this.state.dropdown && this.onDropdownClose(), !n.dropdown && n.dropdown !== this.state.dropdown && this.props.onDropdownOpen();
  }
  componentWillUnmount() {
    this.props.portal && this.props.portal.removeChild(this.dropdownRoot), Ae().removeEventListener(
      "resize",
      Ue(this.updateSelectBounds, this.props.debounceDelay)
    ), Ae().removeEventListener(
      "scroll",
      Ue(this.onScroll, this.props.debounceDelay)
    );
  }
  render() {
    return /* @__PURE__ */ j.jsx(Tt, { onClickOutside: (t) => this.dropDown("close", t), children: /* @__PURE__ */ j.jsxs(
      Jo,
      {
        onKeyDown: this.handleKeyDown,
        "aria-label": "Dropdown select",
        "aria-expanded": this.state.dropdown,
        onClick: (t) => this.dropDown("open", t),
        tabIndex: this.props.disabled ? "-1" : "0",
        direction: this.props.direction,
        style: this.props.style,
        ref: this.select,
        disabled: this.props.disabled,
        className: `${G} ${this.props.className}`,
        color: this.props.color,
        ...this.props.additionalProps,
        children: [
          /* @__PURE__ */ j.jsx(kt, { props: this.props, state: this.state, methods: this.methods }),
          (this.props.name || this.props.required) && /* @__PURE__ */ j.jsx(
            "input",
            {
              tabIndex: -1,
              style: { opacity: 0, width: 0, position: "absolute" },
              name: this.props.name,
              required: this.props.required,
              pattern: this.props.pattern,
              defaultValue: this.state.values.map((t) => t[this.props.labelField]).toString() || [],
              disabled: this.props.disabled
            }
          ),
          this.props.loading && /* @__PURE__ */ j.jsx(Yo, { props: this.props }),
          this.props.clearable && /* @__PURE__ */ j.jsx(zo, { props: this.props, state: this.state, methods: this.methods }),
          this.props.separator && /* @__PURE__ */ j.jsx(Uo, { props: this.props, state: this.state, methods: this.methods }),
          this.props.dropdownHandle && /* @__PURE__ */ j.jsx(
            Ko,
            {
              onClick: () => this.select.current.focus(),
              props: this.props,
              state: this.state,
              methods: this.methods
            }
          ),
          this.state.dropdown && !this.props.disabled && this.renderDropdown()
        ]
      }
    ) });
  }
}
F(Nt, "propTypes", At);
Nt.defaultProps = {
  addPlaceholder: "",
  additionalProps: null,
  autoFocus: !1,
  backspaceDelete: !0,
  clearAllLabel: "Clear all",
  clearOnBlur: !0,
  clearOnSelect: !0,
  clearable: !1,
  closeOnScroll: !1,
  closeOnSelect: !1,
  closeOnClickInput: !1,
  color: "#0074D9",
  compareValuesFunc: To,
  create: !1,
  createNewLabel: "add {search}",
  debounceDelay: 0,
  direction: "ltr",
  disabled: !1,
  disabledLabel: "disabled",
  dropdownGap: 5,
  dropdownHandle: !0,
  dropdownHeight: "300px",
  dropdownPosition: "bottom",
  handleKeyDownFn: () => {
  },
  keepOpen: !1,
  keepSelectedInList: !0,
  labelField: "label",
  loading: !1,
  multi: !1,
  name: null,
  noDataLabel: "No data",
  onChange: () => {
  },
  onClearAll: () => {
  },
  onCreateNew: () => {
  },
  onDropdownClose: () => {
  },
  onDropdownCloseRequest: void 0,
  onDropdownOpen: () => {
  },
  onSelectAll: () => {
  },
  options: [],
  pattern: void 0,
  placeholder: "Select...",
  portal: null,
  required: !1,
  searchBy: "label",
  searchFn: () => {
  },
  searchable: !0,
  selectAll: !1,
  selectAllLabel: "Select all",
  separator: !1,
  sortBy: null,
  valueField: "value",
  values: [],
  defaultMenuIsOpen: !1
};
const Jo = ne.div`
  box-sizing: border-box;
  position: relative;
  display: flex;
  border: 1px solid #ccc;
  width: 100%;
  border-radius: 2px;
  padding: 2px 5px;
  flex-direction: row;
  direction: ${({ direction: e }) => e};
  align-items: center;
  cursor: pointer;
  min-height: 36px;

  ${({ disabled: e }) => e ? "cursor: not-allowed;pointer-events: none;opacity: 0.3;" : "pointer-events: all;"}
  :hover,
  :focus-within {
    border-color: ${({ color: e }) => e};
  }

  :focus,
  :focus-within {
    outline: 0;
    box-shadow: 0 0 0 3px ${({ color: e }) => Ne(e, 0.2)};
  }

  * {
    box-sizing: border-box;
  }
`;
export {
  Nt as Select,
  Nt as default
};
